{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import pandas\n",
    "import numpy\n",
    "import matplotlib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "adata = pandas.read_table(open(\"data/crispr_guide_data.tsv\", \"r\"),\\\n",
    "                          dtype={'gene_name': numpy.str})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>spacer_id</th>\n",
       "      <th>spacer_seq</th>\n",
       "      <th>norm_count_plasmid</th>\n",
       "      <th>norm_count_D7_Rep1</th>\n",
       "      <th>norm_count_D7_Rep2</th>\n",
       "      <th>norm_count_D14_Rep1</th>\n",
       "      <th>norm_count_D14_Rep2</th>\n",
       "      <th>norm_count_PLX7_Rep1</th>\n",
       "      <th>norm_count_PLX7_Rep2</th>\n",
       "      <th>norm_count_PLX14_Rep1</th>\n",
       "      <th>norm_count_PLX14_Rep2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>s_1</td>\n",
       "      <td>ACCTGTAGTTGCCGGCGTGC</td>\n",
       "      <td>10.366767</td>\n",
       "      <td>8.142715</td>\n",
       "      <td>10.177020</td>\n",
       "      <td>7.986137</td>\n",
       "      <td>6.461769</td>\n",
       "      <td>12.260794</td>\n",
       "      <td>9.095489</td>\n",
       "      <td>4.285777</td>\n",
       "      <td>2.816931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>s_2</td>\n",
       "      <td>GTCGCTGAGCTCCGATTCGA</td>\n",
       "      <td>8.805639</td>\n",
       "      <td>9.941240</td>\n",
       "      <td>8.059246</td>\n",
       "      <td>8.605162</td>\n",
       "      <td>6.461769</td>\n",
       "      <td>7.491095</td>\n",
       "      <td>10.515751</td>\n",
       "      <td>4.168428</td>\n",
       "      <td>5.239506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>s_3</td>\n",
       "      <td>AAGTTGCTTGATTGCATTCT</td>\n",
       "      <td>19.493360</td>\n",
       "      <td>26.025195</td>\n",
       "      <td>25.203128</td>\n",
       "      <td>33.277722</td>\n",
       "      <td>21.514938</td>\n",
       "      <td>19.253963</td>\n",
       "      <td>27.913951</td>\n",
       "      <td>9.801189</td>\n",
       "      <td>9.327602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>s_4</td>\n",
       "      <td>ACTTCCCCGAGACCTTTTTG</td>\n",
       "      <td>20.454054</td>\n",
       "      <td>23.147554</td>\n",
       "      <td>23.387894</td>\n",
       "      <td>28.856116</td>\n",
       "      <td>20.649048</td>\n",
       "      <td>20.688459</td>\n",
       "      <td>21.167710</td>\n",
       "      <td>7.219507</td>\n",
       "      <td>8.822898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>s_5</td>\n",
       "      <td>AGTGCCGCTCAATCCATCCC</td>\n",
       "      <td>40.868803</td>\n",
       "      <td>54.801599</td>\n",
       "      <td>54.952807</td>\n",
       "      <td>56.270072</td>\n",
       "      <td>53.353057</td>\n",
       "      <td>51.996336</td>\n",
       "      <td>47.868622</td>\n",
       "      <td>12.206848</td>\n",
       "      <td>13.768989</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_name spacer_id            spacer_seq  norm_count_plasmid  \\\n",
       "0      A1BG       s_1  ACCTGTAGTTGCCGGCGTGC           10.366767   \n",
       "1      A1BG       s_2  GTCGCTGAGCTCCGATTCGA            8.805639   \n",
       "2      A1CF       s_3  AAGTTGCTTGATTGCATTCT           19.493360   \n",
       "3      A1CF       s_4  ACTTCCCCGAGACCTTTTTG           20.454054   \n",
       "4      A1CF       s_5  AGTGCCGCTCAATCCATCCC           40.868803   \n",
       "\n",
       "   norm_count_D7_Rep1  norm_count_D7_Rep2  norm_count_D14_Rep1  \\\n",
       "0            8.142715           10.177020             7.986137   \n",
       "1            9.941240            8.059246             8.605162   \n",
       "2           26.025195           25.203128            33.277722   \n",
       "3           23.147554           23.387894            28.856116   \n",
       "4           54.801599           54.952807            56.270072   \n",
       "\n",
       "   norm_count_D14_Rep2  norm_count_PLX7_Rep1  norm_count_PLX7_Rep2  \\\n",
       "0             6.461769             12.260794              9.095489   \n",
       "1             6.461769              7.491095             10.515751   \n",
       "2            21.514938             19.253963             27.913951   \n",
       "3            20.649048             20.688459             21.167710   \n",
       "4            53.353057             51.996336             47.868622   \n",
       "\n",
       "   norm_count_PLX14_Rep1  norm_count_PLX14_Rep2  \n",
       "0               4.285777               2.816931  \n",
       "1               4.168428               5.239506  \n",
       "2               9.801189               9.327602  \n",
       "3               7.219507               8.822898  \n",
       "4              12.206848              13.768989  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Downloaded log10 data from GSM952580. Manually removed lines from file header to make pandas parseable. Did same for array ID key table. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python3.4/dist-packages/IPython/core/interactiveshell.py:2723: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "#### Trying from log10 values for each gene, as downloaded table:\n",
    "cnvratios = pandas.read_table(open(\"data/GSM952580.txt\", \"r\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID_REF</th>\n",
       "      <th>VALUE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>-0.119148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>0.088217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>0.201403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>-0.321670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>-0.047035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>-0.129696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>-0.101871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>0.037674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>-0.023828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12</td>\n",
       "      <td>-0.040794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>13</td>\n",
       "      <td>0.184694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14</td>\n",
       "      <td>-0.163921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15</td>\n",
       "      <td>-0.028116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>16</td>\n",
       "      <td>0.171655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>17</td>\n",
       "      <td>-0.357686</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>18</td>\n",
       "      <td>-0.921459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19</td>\n",
       "      <td>0.008845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20</td>\n",
       "      <td>-0.215988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>21</td>\n",
       "      <td>0.015971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>-0.022479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>23</td>\n",
       "      <td>0.024846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>24</td>\n",
       "      <td>-0.133274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>25</td>\n",
       "      <td>-0.011583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>26</td>\n",
       "      <td>-0.394723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>27</td>\n",
       "      <td>-0.186855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>28</td>\n",
       "      <td>-0.088957</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>29</td>\n",
       "      <td>-0.192335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>30</td>\n",
       "      <td>0.079182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420259</th>\n",
       "      <td>420260</td>\n",
       "      <td>-0.011781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420260</th>\n",
       "      <td>420261</td>\n",
       "      <td>-0.163962</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420261</th>\n",
       "      <td>420262</td>\n",
       "      <td>0.013428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420262</th>\n",
       "      <td>420263</td>\n",
       "      <td>-0.349211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420263</th>\n",
       "      <td>420264</td>\n",
       "      <td>0.009494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420264</th>\n",
       "      <td>420265</td>\n",
       "      <td>-0.087901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420265</th>\n",
       "      <td>420266</td>\n",
       "      <td>-0.207795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420266</th>\n",
       "      <td>420267</td>\n",
       "      <td>0.430097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420267</th>\n",
       "      <td>420268</td>\n",
       "      <td>0.275209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420268</th>\n",
       "      <td>420269</td>\n",
       "      <td>-0.044793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420269</th>\n",
       "      <td>420270</td>\n",
       "      <td>0.226760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420270</th>\n",
       "      <td>420271</td>\n",
       "      <td>-0.084430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420271</th>\n",
       "      <td>420272</td>\n",
       "      <td>0.162984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420272</th>\n",
       "      <td>420273</td>\n",
       "      <td>0.215579</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420273</th>\n",
       "      <td>420274</td>\n",
       "      <td>-0.094996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420274</th>\n",
       "      <td>420275</td>\n",
       "      <td>-0.164149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420275</th>\n",
       "      <td>420276</td>\n",
       "      <td>0.177670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420276</th>\n",
       "      <td>420277</td>\n",
       "      <td>-0.137954</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420277</th>\n",
       "      <td>420278</td>\n",
       "      <td>0.060273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420278</th>\n",
       "      <td>420279</td>\n",
       "      <td>-0.071245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420279</th>\n",
       "      <td>420280</td>\n",
       "      <td>-0.375589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420280</th>\n",
       "      <td>420281</td>\n",
       "      <td>-0.277492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420281</th>\n",
       "      <td>420282</td>\n",
       "      <td>0.287734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420282</th>\n",
       "      <td>420283</td>\n",
       "      <td>0.185223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420283</th>\n",
       "      <td>420284</td>\n",
       "      <td>0.289022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420284</th>\n",
       "      <td>420285</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420285</th>\n",
       "      <td>420286</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420286</th>\n",
       "      <td>420287</td>\n",
       "      <td>-0.102818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420287</th>\n",
       "      <td>420288</td>\n",
       "      <td>-0.122899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420288</th>\n",
       "      <td>!sample_table_end</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>420289 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   ID_REF     VALUE\n",
       "0                       1 -0.119148\n",
       "1                       2  0.000000\n",
       "2                       3  0.000000\n",
       "3                       4  0.088217\n",
       "4                       5  0.201403\n",
       "5                       6 -0.321670\n",
       "6                       7 -0.047035\n",
       "7                       8 -0.129696\n",
       "8                       9 -0.101871\n",
       "9                      10  0.037674\n",
       "10                     11 -0.023828\n",
       "11                     12 -0.040794\n",
       "12                     13  0.184694\n",
       "13                     14 -0.163921\n",
       "14                     15 -0.028116\n",
       "15                     16  0.171655\n",
       "16                     17 -0.357686\n",
       "17                     18 -0.921459\n",
       "18                     19  0.008845\n",
       "19                     20 -0.215988\n",
       "20                     21  0.015971\n",
       "21                     22 -0.022479\n",
       "22                     23  0.024846\n",
       "23                     24 -0.133274\n",
       "24                     25 -0.011583\n",
       "25                     26 -0.394723\n",
       "26                     27 -0.186855\n",
       "27                     28 -0.088957\n",
       "28                     29 -0.192335\n",
       "29                     30  0.079182\n",
       "...                   ...       ...\n",
       "420259             420260 -0.011781\n",
       "420260             420261 -0.163962\n",
       "420261             420262  0.013428\n",
       "420262             420263 -0.349211\n",
       "420263             420264  0.009494\n",
       "420264             420265 -0.087901\n",
       "420265             420266 -0.207795\n",
       "420266             420267  0.430097\n",
       "420267             420268  0.275209\n",
       "420268             420269 -0.044793\n",
       "420269             420270  0.226760\n",
       "420270             420271 -0.084430\n",
       "420271             420272  0.162984\n",
       "420272             420273  0.215579\n",
       "420273             420274 -0.094996\n",
       "420274             420275 -0.164149\n",
       "420275             420276  0.177670\n",
       "420276             420277 -0.137954\n",
       "420277             420278  0.060273\n",
       "420278             420279 -0.071245\n",
       "420279             420280 -0.375589\n",
       "420280             420281 -0.277492\n",
       "420281             420282  0.287734\n",
       "420282             420283  0.185223\n",
       "420283             420284  0.289022\n",
       "420284             420285  0.000000\n",
       "420285             420286  0.000000\n",
       "420286             420287 -0.102818\n",
       "420287             420288 -0.122899\n",
       "420288  !sample_table_end       NaN\n",
       "\n",
       "[420289 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnvratios"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python3.4/dist-packages/IPython/core/interactiveshell.py:2723: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "array_index = pandas.read_table(open(\"data/arraykey_GPL9777.txt\", \"r\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>COL</th>\n",
       "      <th>ROW</th>\n",
       "      <th>SPOT_ID</th>\n",
       "      <th>CONTROL_TYPE</th>\n",
       "      <th>GB_ACC</th>\n",
       "      <th>GENE_SYMBOL</th>\n",
       "      <th>GENE_NAME</th>\n",
       "      <th>ACCESSION_STRING</th>\n",
       "      <th>CHROMOSOMAL_LOCATION</th>\n",
       "      <th>CYTOBAND</th>\n",
       "      <th>DESCRIPTION</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>528.0</td>\n",
       "      <td>796.0</td>\n",
       "      <td>HsCGHBrightCorner</td>\n",
       "      <td>pos</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>528.0</td>\n",
       "      <td>794.0</td>\n",
       "      <td>DarkCorner</td>\n",
       "      <td>pos</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>528.0</td>\n",
       "      <td>792.0</td>\n",
       "      <td>DarkCorner</td>\n",
       "      <td>pos</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>528.0</td>\n",
       "      <td>790.0</td>\n",
       "      <td>A_16_P17991140</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_012301</td>\n",
       "      <td>MAGI2</td>\n",
       "      <td>membrane associated guanylate kinase, WW and P...</td>\n",
       "      <td>ref|NM_012301</td>\n",
       "      <td>chr7:77827408-77827467</td>\n",
       "      <td>hs|q21.11</td>\n",
       "      <td>Homo sapiens membrane associated guanylate kin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>528.0</td>\n",
       "      <td>788.0</td>\n",
       "      <td>A_16_P33811607</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_207112</td>\n",
       "      <td>HAGHL</td>\n",
       "      <td>hydroxyacylglutathione hydrolase-like</td>\n",
       "      <td>ref|NM_207112|ref|NM_032304</td>\n",
       "      <td>chr16:718810-718854</td>\n",
       "      <td>hs|p13.3</td>\n",
       "      <td>Homo sapiens hydroxyacylglutathione hydrolase-...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ID    COL    ROW            SPOT_ID CONTROL_TYPE     GB_ACC GENE_SYMBOL  \\\n",
       "0  1  528.0  796.0  HsCGHBrightCorner          pos        NaN         NaN   \n",
       "1  2  528.0  794.0         DarkCorner          pos        NaN         NaN   \n",
       "2  3  528.0  792.0         DarkCorner          pos        NaN         NaN   \n",
       "3  4  528.0  790.0     A_16_P17991140        FALSE  NM_012301       MAGI2   \n",
       "4  5  528.0  788.0     A_16_P33811607        FALSE  NM_207112       HAGHL   \n",
       "\n",
       "                                           GENE_NAME  \\\n",
       "0                                                NaN   \n",
       "1                                                NaN   \n",
       "2                                                NaN   \n",
       "3  membrane associated guanylate kinase, WW and P...   \n",
       "4              hydroxyacylglutathione hydrolase-like   \n",
       "\n",
       "              ACCESSION_STRING    CHROMOSOMAL_LOCATION   CYTOBAND  \\\n",
       "0                          NaN                     NaN        NaN   \n",
       "1                          NaN                     NaN        NaN   \n",
       "2                          NaN                     NaN        NaN   \n",
       "3                ref|NM_012301  chr7:77827408-77827467  hs|q21.11   \n",
       "4  ref|NM_207112|ref|NM_032304     chr16:718810-718854   hs|p13.3   \n",
       "\n",
       "                                         DESCRIPTION  \n",
       "0                                                NaN  \n",
       "1                                                NaN  \n",
       "2                                                NaN  \n",
       "3  Homo sapiens membrane associated guanylate kin...  \n",
       "4  Homo sapiens hydroxyacylglutathione hydrolase-...  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "array_index.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cnv_ratios_and_genenames = pandas.concat([cnvratios, array_index], axis=1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ID_REF                                  1\n",
       "VALUE                           -0.119148\n",
       "ID                                      1\n",
       "COL                                   528\n",
       "ROW                                   796\n",
       "SPOT_ID                 HsCGHBrightCorner\n",
       "CONTROL_TYPE                          pos\n",
       "GB_ACC                                NaN\n",
       "GENE_SYMBOL                           NaN\n",
       "GENE_NAME                             NaN\n",
       "ACCESSION_STRING                      NaN\n",
       "CHROMOSOMAL_LOCATION                  NaN\n",
       "CYTOBAND                              NaN\n",
       "DESCRIPTION                           NaN\n",
       "Name: 0, dtype: object"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv_ratios_and_genenames.iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([1, -0.11914833429999999, 1, 528.0, 796.0, 'HsCGHBrightCorner',\n",
       "       'pos', nan, nan, nan, nan, nan, nan, nan], dtype=object)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv_ratios_and_genenames.values[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID_REF                    !sample_table_end\n",
      "VALUE                                   NaN\n",
      "ID                      !platform_table_end\n",
      "COL                                     NaN\n",
      "ROW                                     NaN\n",
      "SPOT_ID                                 NaN\n",
      "CONTROL_TYPE                            NaN\n",
      "GB_ACC                                  NaN\n",
      "GENE_SYMBOL                             NaN\n",
      "GENE_NAME                               NaN\n",
      "ACCESSION_STRING                        NaN\n",
      "CHROMOSOMAL_LOCATION                    NaN\n",
      "CYTOBAND                                NaN\n",
      "DESCRIPTION                             NaN\n",
      "Name: 420288, dtype: object\n"
     ]
    }
   ],
   "source": [
    "# Print then remove the last line from the table; was just a footer\n",
    "print(cnv_ratios_and_genenames.iloc[-1])\n",
    "cnv_ratios_and_genenames = cnv_ratios_and_genenames.iloc[:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def check_ids_in_contantenated_table_match():\n",
    "\n",
    "    for item in cnv_ratios_and_genenames.iterrows():\n",
    "        if item[1][\"ID_REF\"] == item[1][\"ID\"]:\n",
    "            None\n",
    "        else:\n",
    "            print(item[1])\n",
    "            print(\"\\n\")\n",
    "            print(repr(item[1][\"ID_REF\"]))\n",
    "            print(repr(item[1][\"ID\"]))\n",
    "            break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID_REF                                   262145\n",
      "VALUE                                -0.0102334\n",
      "ID                                       262145\n",
      "COL                                         199\n",
      "ROW                                         276\n",
      "SPOT_ID                          A_18_P20006280\n",
      "CONTROL_TYPE                              FALSE\n",
      "GB_ACC                                      NaN\n",
      "GENE_SYMBOL                                 NaN\n",
      "GENE_NAME                                   NaN\n",
      "ACCESSION_STRING                            NaN\n",
      "CHROMOSOMAL_LOCATION    chr14:44242718-44242777\n",
      "CYTOBAND                               hs|q21.3\n",
      "DESCRIPTION                                 NaN\n",
      "Name: 262144, dtype: object\n",
      "\n",
      "\n",
      "'262145'\n",
      "262145\n"
     ]
    }
   ],
   "source": [
    "check_ids_in_contantenated_table_match()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": false
   },
   "source": [
    "Hmm. They don't match; found that at some point the Pandas import decided to take the ID values as strings instead of ints. Make sure they're all ints."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "cnv_ratios_and_genenames[\"ID\"] = cnv_ratios_and_genenames[\"ID\"].apply(int)\n",
    "cnv_ratios_and_genenames[\"ID_REF\"] = cnv_ratios_and_genenames[\"ID_REF\"].apply(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "check_ids_in_contantenated_table_match()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "OK; look good now. So, we have CNV data with gene names/IDs associated. Next take the guide data and extract that subset of the CNV data that overlaps with the guide data. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>spacer_id</th>\n",
       "      <th>spacer_seq</th>\n",
       "      <th>norm_count_plasmid</th>\n",
       "      <th>norm_count_D7_Rep1</th>\n",
       "      <th>norm_count_D7_Rep2</th>\n",
       "      <th>norm_count_D14_Rep1</th>\n",
       "      <th>norm_count_D14_Rep2</th>\n",
       "      <th>norm_count_PLX7_Rep1</th>\n",
       "      <th>norm_count_PLX7_Rep2</th>\n",
       "      <th>norm_count_PLX14_Rep1</th>\n",
       "      <th>norm_count_PLX14_Rep2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>s_1</td>\n",
       "      <td>ACCTGTAGTTGCCGGCGTGC</td>\n",
       "      <td>10.366767</td>\n",
       "      <td>8.142715</td>\n",
       "      <td>10.177020</td>\n",
       "      <td>7.986137</td>\n",
       "      <td>6.461769</td>\n",
       "      <td>12.260794</td>\n",
       "      <td>9.095489</td>\n",
       "      <td>4.285777</td>\n",
       "      <td>2.816931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>s_2</td>\n",
       "      <td>GTCGCTGAGCTCCGATTCGA</td>\n",
       "      <td>8.805639</td>\n",
       "      <td>9.941240</td>\n",
       "      <td>8.059246</td>\n",
       "      <td>8.605162</td>\n",
       "      <td>6.461769</td>\n",
       "      <td>7.491095</td>\n",
       "      <td>10.515751</td>\n",
       "      <td>4.168428</td>\n",
       "      <td>5.239506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>s_3</td>\n",
       "      <td>AAGTTGCTTGATTGCATTCT</td>\n",
       "      <td>19.493360</td>\n",
       "      <td>26.025195</td>\n",
       "      <td>25.203128</td>\n",
       "      <td>33.277722</td>\n",
       "      <td>21.514938</td>\n",
       "      <td>19.253963</td>\n",
       "      <td>27.913951</td>\n",
       "      <td>9.801189</td>\n",
       "      <td>9.327602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>s_4</td>\n",
       "      <td>ACTTCCCCGAGACCTTTTTG</td>\n",
       "      <td>20.454054</td>\n",
       "      <td>23.147554</td>\n",
       "      <td>23.387894</td>\n",
       "      <td>28.856116</td>\n",
       "      <td>20.649048</td>\n",
       "      <td>20.688459</td>\n",
       "      <td>21.167710</td>\n",
       "      <td>7.219507</td>\n",
       "      <td>8.822898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>s_5</td>\n",
       "      <td>AGTGCCGCTCAATCCATCCC</td>\n",
       "      <td>40.868803</td>\n",
       "      <td>54.801599</td>\n",
       "      <td>54.952807</td>\n",
       "      <td>56.270072</td>\n",
       "      <td>53.353057</td>\n",
       "      <td>51.996336</td>\n",
       "      <td>47.868622</td>\n",
       "      <td>12.206848</td>\n",
       "      <td>13.768989</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_name spacer_id            spacer_seq  norm_count_plasmid  \\\n",
       "0      A1BG       s_1  ACCTGTAGTTGCCGGCGTGC           10.366767   \n",
       "1      A1BG       s_2  GTCGCTGAGCTCCGATTCGA            8.805639   \n",
       "2      A1CF       s_3  AAGTTGCTTGATTGCATTCT           19.493360   \n",
       "3      A1CF       s_4  ACTTCCCCGAGACCTTTTTG           20.454054   \n",
       "4      A1CF       s_5  AGTGCCGCTCAATCCATCCC           40.868803   \n",
       "\n",
       "   norm_count_D7_Rep1  norm_count_D7_Rep2  norm_count_D14_Rep1  \\\n",
       "0            8.142715           10.177020             7.986137   \n",
       "1            9.941240            8.059246             8.605162   \n",
       "2           26.025195           25.203128            33.277722   \n",
       "3           23.147554           23.387894            28.856116   \n",
       "4           54.801599           54.952807            56.270072   \n",
       "\n",
       "   norm_count_D14_Rep2  norm_count_PLX7_Rep1  norm_count_PLX7_Rep2  \\\n",
       "0             6.461769             12.260794              9.095489   \n",
       "1             6.461769              7.491095             10.515751   \n",
       "2            21.514938             19.253963             27.913951   \n",
       "3            20.649048             20.688459             21.167710   \n",
       "4            53.353057             51.996336             47.868622   \n",
       "\n",
       "   norm_count_PLX14_Rep1  norm_count_PLX14_Rep2  \n",
       "0               4.285777               2.816931  \n",
       "1               4.168428               5.239506  \n",
       "2               9.801189               9.327602  \n",
       "3               7.219507               8.822898  \n",
       "4              12.206848              13.768989  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "At this point we need to take the CNV data that intersects with regions covered by guides. The ideal here would to be map each guide back to genomic coordinates and correlate those with the coordinates supplied in the CGH array data. However, for the sake of completing this in a reasonable amount of time, I'm going to trust that the `gene_name` in the guide data can be intersected with the `GENE_SYMBOL` in the CNV data.\n",
    "\n",
    "So, next we take the CNV data entries that have `GENE_SYMBOL` set (i.e. not an `NaN`). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "cnv_ratios_and_genenames = cnv_ratios_and_genenames.dropna(subset=[\"GENE_SYMBOL\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The next step relies of being able to intersect the two datasets; ideally all the gene names in the sgRNA data are represented in the CGH data. We're also going to proceed differently depending on whether genes in the CGH dataset are represented in more than one array spot. This will bring us to the first question of how to statistically represent multiple mesurements in both the sgRNAs and CNV spots within a gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID_REF</th>\n",
       "      <th>VALUE</th>\n",
       "      <th>ID</th>\n",
       "      <th>COL</th>\n",
       "      <th>ROW</th>\n",
       "      <th>SPOT_ID</th>\n",
       "      <th>CONTROL_TYPE</th>\n",
       "      <th>GB_ACC</th>\n",
       "      <th>GENE_SYMBOL</th>\n",
       "      <th>GENE_NAME</th>\n",
       "      <th>ACCESSION_STRING</th>\n",
       "      <th>CHROMOSOMAL_LOCATION</th>\n",
       "      <th>CYTOBAND</th>\n",
       "      <th>DESCRIPTION</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>235901</th>\n",
       "      <td>235902</td>\n",
       "      <td>-0.059363</td>\n",
       "      <td>235902</td>\n",
       "      <td>232.0</td>\n",
       "      <td>226.0</td>\n",
       "      <td>A_16_P41237987</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_130786</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>alpha-1-B glycoprotein</td>\n",
       "      <td>ref|NM_130786</td>\n",
       "      <td>chr19:63551904-63551963</td>\n",
       "      <td>hs|q13.43</td>\n",
       "      <td>Homo sapiens alpha-1-B glycoprotein (A1BG), mRNA.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>332477</th>\n",
       "      <td>332478</td>\n",
       "      <td>0.013735</td>\n",
       "      <td>332478</td>\n",
       "      <td>111.0</td>\n",
       "      <td>501.0</td>\n",
       "      <td>A_14_P101758</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52315385-52315441</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184747</th>\n",
       "      <td>184748</td>\n",
       "      <td>-0.266349</td>\n",
       "      <td>184748</td>\n",
       "      <td>296.0</td>\n",
       "      <td>646.0</td>\n",
       "      <td>A_18_P10736965</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52310289-52310348</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227995</th>\n",
       "      <td>227996</td>\n",
       "      <td>-0.134579</td>\n",
       "      <td>227996</td>\n",
       "      <td>242.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>A_16_P18922462</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52245714-52245773</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242555</th>\n",
       "      <td>242556</td>\n",
       "      <td>-0.333080</td>\n",
       "      <td>242556</td>\n",
       "      <td>224.0</td>\n",
       "      <td>449.0</td>\n",
       "      <td>A_16_P18922623</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52301417-52301476</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88687</th>\n",
       "      <td>88688</td>\n",
       "      <td>-0.193421</td>\n",
       "      <td>88688</td>\n",
       "      <td>417.0</td>\n",
       "      <td>134.0</td>\n",
       "      <td>A_16_P39030680</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52271703-52271762</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242527</th>\n",
       "      <td>242528</td>\n",
       "      <td>-0.078122</td>\n",
       "      <td>242528</td>\n",
       "      <td>224.0</td>\n",
       "      <td>505.0</td>\n",
       "      <td>A_16_P02260125</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52285879-52285938</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271992</th>\n",
       "      <td>271993</td>\n",
       "      <td>0.022711</td>\n",
       "      <td>271993</td>\n",
       "      <td>187.0</td>\n",
       "      <td>479.0</td>\n",
       "      <td>A_16_P02260134</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52289698-52289757</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182310</th>\n",
       "      <td>182311</td>\n",
       "      <td>-0.148692</td>\n",
       "      <td>182311</td>\n",
       "      <td>299.0</td>\n",
       "      <td>744.0</td>\n",
       "      <td>A_16_P39030652</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52262400-52262459</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53236</th>\n",
       "      <td>53237</td>\n",
       "      <td>-0.180240</td>\n",
       "      <td>53237</td>\n",
       "      <td>462.0</td>\n",
       "      <td>191.0</td>\n",
       "      <td>A_16_P39030621</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52249696-52249755</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151091</th>\n",
       "      <td>151092</td>\n",
       "      <td>-0.134324</td>\n",
       "      <td>151092</td>\n",
       "      <td>339.0</td>\n",
       "      <td>297.0</td>\n",
       "      <td>A_16_P18922487</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52252979-52253038</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97854</th>\n",
       "      <td>97855</td>\n",
       "      <td>-0.101267</td>\n",
       "      <td>97855</td>\n",
       "      <td>406.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>A_16_P18922549</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52278706-52278765</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66678</th>\n",
       "      <td>66679</td>\n",
       "      <td>-0.382034</td>\n",
       "      <td>66679</td>\n",
       "      <td>445.0</td>\n",
       "      <td>371.0</td>\n",
       "      <td>A_16_P18922597</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52292685-52292744</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28487</th>\n",
       "      <td>28488</td>\n",
       "      <td>-0.094625</td>\n",
       "      <td>28488</td>\n",
       "      <td>493.0</td>\n",
       "      <td>337.0</td>\n",
       "      <td>A_16_P32662933</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52240955-52241014</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36946</th>\n",
       "      <td>36947</td>\n",
       "      <td>-0.166890</td>\n",
       "      <td>36947</td>\n",
       "      <td>482.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>A_16_P39030769</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52298312-52298371</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382524</th>\n",
       "      <td>382525</td>\n",
       "      <td>-0.184075</td>\n",
       "      <td>382525</td>\n",
       "      <td>48.0</td>\n",
       "      <td>703.0</td>\n",
       "      <td>A_16_P18922640</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52306765-52306824</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339939</th>\n",
       "      <td>339940</td>\n",
       "      <td>-0.093597</td>\n",
       "      <td>339940</td>\n",
       "      <td>101.0</td>\n",
       "      <td>702.0</td>\n",
       "      <td>A_14_P115717</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52265960-52266019</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284504</th>\n",
       "      <td>284505</td>\n",
       "      <td>-0.051589</td>\n",
       "      <td>284505</td>\n",
       "      <td>171.0</td>\n",
       "      <td>132.0</td>\n",
       "      <td>A_16_P39030646</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52258030-52258089</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174046</th>\n",
       "      <td>174047</td>\n",
       "      <td>-0.175303</td>\n",
       "      <td>174047</td>\n",
       "      <td>310.0</td>\n",
       "      <td>555.0</td>\n",
       "      <td>A_14_P127769</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_014576</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>ref|NM_014576|ref|NM_138932|ref|NM_138933</td>\n",
       "      <td>chr10:52236360-52236419</td>\n",
       "      <td>hs|q11.23</td>\n",
       "      <td>Homo sapiens APOBEC1 complementation factor (A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49694</th>\n",
       "      <td>49695</td>\n",
       "      <td>-0.016057</td>\n",
       "      <td>49695</td>\n",
       "      <td>466.0</td>\n",
       "      <td>112.0</td>\n",
       "      <td>A_16_P03110179</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_001142333</td>\n",
       "      <td>A2BP1</td>\n",
       "      <td>ataxin 2-binding protein 1</td>\n",
       "      <td>ref|NM_001142333|ref|NM_018723|ref|NM_00114233...</td>\n",
       "      <td>chr16:7387703-7387762</td>\n",
       "      <td>hs|p13.2</td>\n",
       "      <td>Homo sapiens ataxin 2-binding protein 1 (A2BP1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119756</th>\n",
       "      <td>119757</td>\n",
       "      <td>0.058620</td>\n",
       "      <td>119757</td>\n",
       "      <td>378.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>A_16_P20390559</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_001142333</td>\n",
       "      <td>A2BP1</td>\n",
       "      <td>ataxin 2-binding protein 1</td>\n",
       "      <td>ref|NM_001142333|ref|NM_018723</td>\n",
       "      <td>chr16:6634165-6634224</td>\n",
       "      <td>hs|p13.2</td>\n",
       "      <td>Homo sapiens ataxin 2-binding protein 1 (A2BP1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14971</th>\n",
       "      <td>14972</td>\n",
       "      <td>-0.131744</td>\n",
       "      <td>14972</td>\n",
       "      <td>510.0</td>\n",
       "      <td>305.0</td>\n",
       "      <td>A_16_P40547064</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_001142333</td>\n",
       "      <td>A2BP1</td>\n",
       "      <td>ataxin 2-binding protein 1</td>\n",
       "      <td>ref|NM_001142333|ref|NM_018723</td>\n",
       "      <td>chr16:6609413-6609472</td>\n",
       "      <td>hs|p13.2</td>\n",
       "      <td>Homo sapiens ataxin 2-binding protein 1 (A2BP1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306127</th>\n",
       "      <td>306128</td>\n",
       "      <td>0.071577</td>\n",
       "      <td>306128</td>\n",
       "      <td>144.0</td>\n",
       "      <td>665.0</td>\n",
       "      <td>A_18_P12377530</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_001142333</td>\n",
       "      <td>A2BP1</td>\n",
       "      <td>ataxin 2-binding protein 1</td>\n",
       "      <td>ref|NM_001142333|ref|NM_018723</td>\n",
       "      <td>chr16:6159042-6159101</td>\n",
       "      <td>hs|p13.3</td>\n",
       "      <td>Homo sapiens ataxin 2-binding protein 1 (A2BP1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340231</th>\n",
       "      <td>340232</td>\n",
       "      <td>-0.125075</td>\n",
       "      <td>340232</td>\n",
       "      <td>101.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>A_16_P20392765</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_001142333</td>\n",
       "      <td>A2BP1</td>\n",
       "      <td>ataxin 2-binding protein 1</td>\n",
       "      <td>ref|NM_001142333|ref|NM_018723|ref|NM_00114233...</td>\n",
       "      <td>chr16:7449772-7449831</td>\n",
       "      <td>hs|p13.2</td>\n",
       "      <td>Homo sapiens ataxin 2-binding protein 1 (A2BP1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393200</th>\n",
       "      <td>393201</td>\n",
       "      <td>0.045828</td>\n",
       "      <td>393201</td>\n",
       "      <td>35.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>A_16_P03109077</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_001142333</td>\n",
       "      <td>A2BP1</td>\n",
       "      <td>ataxin 2-binding protein 1</td>\n",
       "      <td>ref|NM_001142333|ref|NM_018723</td>\n",
       "      <td>chr16:6719675-6719734</td>\n",
       "      <td>hs|p13.2</td>\n",
       "      <td>Homo sapiens ataxin 2-binding protein 1 (A2BP1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285713</th>\n",
       "      <td>285714</td>\n",
       "      <td>-0.028778</td>\n",
       "      <td>285714</td>\n",
       "      <td>170.0</td>\n",
       "      <td>101.0</td>\n",
       "      <td>A_18_P12380328</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_001142333</td>\n",
       "      <td>A2BP1</td>\n",
       "      <td>ataxin 2-binding protein 1</td>\n",
       "      <td>ref|NM_001142333|ref|NM_018723|ref|NM_001142334</td>\n",
       "      <td>chr16:7166984-7167043</td>\n",
       "      <td>hs|p13.2</td>\n",
       "      <td>Homo sapiens ataxin 2-binding protein 1 (A2BP1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63012</th>\n",
       "      <td>63013</td>\n",
       "      <td>-0.155494</td>\n",
       "      <td>63013</td>\n",
       "      <td>449.0</td>\n",
       "      <td>540.0</td>\n",
       "      <td>A_16_P20391759</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_001142333</td>\n",
       "      <td>A2BP1</td>\n",
       "      <td>ataxin 2-binding protein 1</td>\n",
       "      <td>ref|NM_001142333|ref|NM_018723|ref|NM_001142334</td>\n",
       "      <td>chr16:7101613-7101672</td>\n",
       "      <td>hs|p13.2</td>\n",
       "      <td>Homo sapiens ataxin 2-binding protein 1 (A2BP1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368379</th>\n",
       "      <td>368380</td>\n",
       "      <td>0.048246</td>\n",
       "      <td>368380</td>\n",
       "      <td>66.0</td>\n",
       "      <td>337.0</td>\n",
       "      <td>A_16_P03108492</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_001142333</td>\n",
       "      <td>A2BP1</td>\n",
       "      <td>ataxin 2-binding protein 1</td>\n",
       "      <td>ref|NM_001142333|ref|NM_018723</td>\n",
       "      <td>chr16:6364057-6364116</td>\n",
       "      <td>hs|p13.2</td>\n",
       "      <td>Homo sapiens ataxin 2-binding protein 1 (A2BP1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178940</th>\n",
       "      <td>178941</td>\n",
       "      <td>0.173739</td>\n",
       "      <td>178941</td>\n",
       "      <td>304.0</td>\n",
       "      <td>319.0</td>\n",
       "      <td>A_18_P12378393</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_001142333</td>\n",
       "      <td>A2BP1</td>\n",
       "      <td>ataxin 2-binding protein 1</td>\n",
       "      <td>ref|NM_001142333|ref|NM_018723</td>\n",
       "      <td>chr16:6647650-6647709</td>\n",
       "      <td>hs|p13.2</td>\n",
       "      <td>Homo sapiens ataxin 2-binding protein 1 (A2BP1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260875</th>\n",
       "      <td>260876</td>\n",
       "      <td>0.097303</td>\n",
       "      <td>260876</td>\n",
       "      <td>201.0</td>\n",
       "      <td>425.0</td>\n",
       "      <td>A_18_P12376163</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_001142333</td>\n",
       "      <td>A2BP1</td>\n",
       "      <td>ataxin 2-binding protein 1</td>\n",
       "      <td>ref|NM_001142333|ref|NM_018723</td>\n",
       "      <td>chr16:6198892-6198951</td>\n",
       "      <td>hs|p13.3</td>\n",
       "      <td>Homo sapiens ataxin 2-binding protein 1 (A2BP1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304068</th>\n",
       "      <td>304069</td>\n",
       "      <td>-0.154090</td>\n",
       "      <td>304069</td>\n",
       "      <td>147.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>A_16_P15179920</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77899238-77899297</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415770</th>\n",
       "      <td>415771</td>\n",
       "      <td>0.063581</td>\n",
       "      <td>415771</td>\n",
       "      <td>6.0</td>\n",
       "      <td>280.0</td>\n",
       "      <td>A_14_P106772</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77820120-77820179</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124098</th>\n",
       "      <td>124099</td>\n",
       "      <td>0.077319</td>\n",
       "      <td>124099</td>\n",
       "      <td>373.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>A_16_P15179821</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77857751-77857810</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331637</th>\n",
       "      <td>331638</td>\n",
       "      <td>0.237792</td>\n",
       "      <td>331638</td>\n",
       "      <td>112.0</td>\n",
       "      <td>589.0</td>\n",
       "      <td>A_14_P109546</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77819229-77819288</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79674</th>\n",
       "      <td>79675</td>\n",
       "      <td>0.269498</td>\n",
       "      <td>79675</td>\n",
       "      <td>428.0</td>\n",
       "      <td>648.0</td>\n",
       "      <td>A_16_P15179878</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77876492-77876551</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69060</th>\n",
       "      <td>69061</td>\n",
       "      <td>0.185846</td>\n",
       "      <td>69061</td>\n",
       "      <td>442.0</td>\n",
       "      <td>383.0</td>\n",
       "      <td>A_14_P101876</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77903687-77903746</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281338</th>\n",
       "      <td>281339</td>\n",
       "      <td>-0.288873</td>\n",
       "      <td>281339</td>\n",
       "      <td>175.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>A_14_P125607</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77871494-77871553</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47524</th>\n",
       "      <td>47525</td>\n",
       "      <td>0.130944</td>\n",
       "      <td>47525</td>\n",
       "      <td>469.0</td>\n",
       "      <td>471.0</td>\n",
       "      <td>A_16_P15179714</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77807227-77807286</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267528</th>\n",
       "      <td>267529</td>\n",
       "      <td>0.055455</td>\n",
       "      <td>267529</td>\n",
       "      <td>192.0</td>\n",
       "      <td>652.0</td>\n",
       "      <td>A_16_P00101391</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77884802-77884861</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175196</th>\n",
       "      <td>175197</td>\n",
       "      <td>0.173158</td>\n",
       "      <td>175197</td>\n",
       "      <td>308.0</td>\n",
       "      <td>644.0</td>\n",
       "      <td>A_16_P15179888</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77879708-77879767</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54830</th>\n",
       "      <td>54831</td>\n",
       "      <td>0.152616</td>\n",
       "      <td>54831</td>\n",
       "      <td>460.0</td>\n",
       "      <td>187.0</td>\n",
       "      <td>A_16_P15179908</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77894759-77894818</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6950</th>\n",
       "      <td>6951</td>\n",
       "      <td>0.140221</td>\n",
       "      <td>6951</td>\n",
       "      <td>520.0</td>\n",
       "      <td>427.0</td>\n",
       "      <td>A_18_P17656585</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77829907-77829966</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249813</th>\n",
       "      <td>249814</td>\n",
       "      <td>-0.031044</td>\n",
       "      <td>249814</td>\n",
       "      <td>215.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>A_16_P35188503</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77911289-77911348</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227335</th>\n",
       "      <td>227336</td>\n",
       "      <td>-0.053644</td>\n",
       "      <td>227336</td>\n",
       "      <td>243.0</td>\n",
       "      <td>641.0</td>\n",
       "      <td>A_16_P35188381</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77860965-77861024</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51494</th>\n",
       "      <td>51495</td>\n",
       "      <td>-0.134389</td>\n",
       "      <td>51495</td>\n",
       "      <td>464.0</td>\n",
       "      <td>491.0</td>\n",
       "      <td>A_16_P15179845</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77866835-77866894</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132936</th>\n",
       "      <td>132937</td>\n",
       "      <td>-0.075553</td>\n",
       "      <td>132937</td>\n",
       "      <td>361.0</td>\n",
       "      <td>788.0</td>\n",
       "      <td>A_16_P00101412</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77908354-77908413</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89609</th>\n",
       "      <td>89610</td>\n",
       "      <td>0.037540</td>\n",
       "      <td>89610</td>\n",
       "      <td>416.0</td>\n",
       "      <td>677.0</td>\n",
       "      <td>A_16_P15179725</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NM_015534</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>zinc finger, ZZ-type containing 3</td>\n",
       "      <td>ref|NM_015534</td>\n",
       "      <td>chr1:77812118-77812177</td>\n",
       "      <td>hs|p31.1</td>\n",
       "      <td>Homo sapiens zinc finger, ZZ-type containing 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163154</th>\n",
       "      <td>163155</td>\n",
       "      <td>-0.253419</td>\n",
       "      <td>163155</td>\n",
       "      <td>324.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>A_16_P34823706</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NR_001591</td>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>TPTE pseudogene</td>\n",
       "      <td>ref|NR_001591</td>\n",
       "      <td>chr22:15508442-15508499</td>\n",
       "      <td>hs|q11.1</td>\n",
       "      <td>Homo sapiens TPTE pseudogene (psiTPTE22), non-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338273</th>\n",
       "      <td>338274</td>\n",
       "      <td>-0.117498</td>\n",
       "      <td>338274</td>\n",
       "      <td>104.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>A_16_P03598387</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NR_001591</td>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>TPTE pseudogene</td>\n",
       "      <td>ref|NR_001591</td>\n",
       "      <td>chr22:15464955-15465014</td>\n",
       "      <td>hs|q11.1</td>\n",
       "      <td>Homo sapiens TPTE pseudogene (psiTPTE22), non-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96420</th>\n",
       "      <td>96421</td>\n",
       "      <td>-0.038295</td>\n",
       "      <td>96421</td>\n",
       "      <td>407.0</td>\n",
       "      <td>588.0</td>\n",
       "      <td>A_18_P13952367</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NR_001591</td>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>TPTE pseudogene</td>\n",
       "      <td>ref|NR_001591</td>\n",
       "      <td>chr22:15471221-15471273</td>\n",
       "      <td>hs|q11.1</td>\n",
       "      <td>Homo sapiens TPTE pseudogene (psiTPTE22), non-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357092</th>\n",
       "      <td>357093</td>\n",
       "      <td>-0.054389</td>\n",
       "      <td>357093</td>\n",
       "      <td>80.0</td>\n",
       "      <td>623.0</td>\n",
       "      <td>A_14_P115613</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NR_001591</td>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>TPTE pseudogene</td>\n",
       "      <td>ref|NR_001591</td>\n",
       "      <td>chr22:15476855-15476914</td>\n",
       "      <td>hs|q11.1</td>\n",
       "      <td>Homo sapiens TPTE pseudogene (psiTPTE22), non-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47466</th>\n",
       "      <td>47467</td>\n",
       "      <td>-0.046077</td>\n",
       "      <td>47467</td>\n",
       "      <td>469.0</td>\n",
       "      <td>587.0</td>\n",
       "      <td>A_16_P18818145</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NR_026743</td>\n",
       "      <td>tAKR</td>\n",
       "      <td>aldo-keto reductase, truncated</td>\n",
       "      <td>ref|NR_026743</td>\n",
       "      <td>chr10:4934320-4934379</td>\n",
       "      <td>hs|p15.1</td>\n",
       "      <td>Homo sapiens aldo-keto reductase, truncated (t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107044</th>\n",
       "      <td>107045</td>\n",
       "      <td>0.062195</td>\n",
       "      <td>107045</td>\n",
       "      <td>394.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>A_16_P18818149</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NR_026743</td>\n",
       "      <td>tAKR</td>\n",
       "      <td>aldo-keto reductase, truncated</td>\n",
       "      <td>ref|NR_026743</td>\n",
       "      <td>chr10:4939503-4939562</td>\n",
       "      <td>hs|p15.1</td>\n",
       "      <td>Homo sapiens aldo-keto reductase, truncated (t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103252</th>\n",
       "      <td>103253</td>\n",
       "      <td>-0.109646</td>\n",
       "      <td>103253</td>\n",
       "      <td>399.0</td>\n",
       "      <td>455.0</td>\n",
       "      <td>A_16_P18818159</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NR_026743</td>\n",
       "      <td>tAKR</td>\n",
       "      <td>aldo-keto reductase, truncated</td>\n",
       "      <td>ref|NR_026743</td>\n",
       "      <td>chr10:4945256-4945315</td>\n",
       "      <td>hs|p15.1</td>\n",
       "      <td>Homo sapiens aldo-keto reductase, truncated (t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189784</th>\n",
       "      <td>189785</td>\n",
       "      <td>-0.025934</td>\n",
       "      <td>189785</td>\n",
       "      <td>290.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>A_16_P02204406</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NR_026743</td>\n",
       "      <td>tAKR</td>\n",
       "      <td>aldo-keto reductase, truncated</td>\n",
       "      <td>ref|NR_026743</td>\n",
       "      <td>chr10:4909104-4909163</td>\n",
       "      <td>hs|p15.1</td>\n",
       "      <td>Homo sapiens aldo-keto reductase, truncated (t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399584</th>\n",
       "      <td>399585</td>\n",
       "      <td>-0.217337</td>\n",
       "      <td>399585</td>\n",
       "      <td>27.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>A_16_P02204415</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NR_026743</td>\n",
       "      <td>tAKR</td>\n",
       "      <td>aldo-keto reductase, truncated</td>\n",
       "      <td>ref|NR_026743</td>\n",
       "      <td>chr10:4918272-4918331</td>\n",
       "      <td>hs|p15.1</td>\n",
       "      <td>Homo sapiens aldo-keto reductase, truncated (t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328696</th>\n",
       "      <td>328697</td>\n",
       "      <td>-0.059410</td>\n",
       "      <td>328697</td>\n",
       "      <td>116.0</td>\n",
       "      <td>103.0</td>\n",
       "      <td>A_16_P02204419</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NR_026743</td>\n",
       "      <td>tAKR</td>\n",
       "      <td>aldo-keto reductase, truncated</td>\n",
       "      <td>ref|NR_026743</td>\n",
       "      <td>chr10:4920849-4920908</td>\n",
       "      <td>hs|p15.1</td>\n",
       "      <td>Homo sapiens aldo-keto reductase, truncated (t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>269979</th>\n",
       "      <td>269980</td>\n",
       "      <td>-0.085281</td>\n",
       "      <td>269980</td>\n",
       "      <td>189.0</td>\n",
       "      <td>526.0</td>\n",
       "      <td>A_16_P02204427</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NR_026743</td>\n",
       "      <td>tAKR</td>\n",
       "      <td>aldo-keto reductase, truncated</td>\n",
       "      <td>ref|NR_026743</td>\n",
       "      <td>chr10:4924101-4924160</td>\n",
       "      <td>hs|p15.1</td>\n",
       "      <td>Homo sapiens aldo-keto reductase, truncated (t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197213</th>\n",
       "      <td>197214</td>\n",
       "      <td>0.036307</td>\n",
       "      <td>197214</td>\n",
       "      <td>281.0</td>\n",
       "      <td>389.0</td>\n",
       "      <td>A_16_P02204398</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NR_026743</td>\n",
       "      <td>tAKR</td>\n",
       "      <td>aldo-keto reductase, truncated</td>\n",
       "      <td>ref|NR_026743</td>\n",
       "      <td>chr10:4904018-4904077</td>\n",
       "      <td>hs|p15.1</td>\n",
       "      <td>Homo sapiens aldo-keto reductase, truncated (t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356671</th>\n",
       "      <td>356672</td>\n",
       "      <td>-0.084985</td>\n",
       "      <td>356672</td>\n",
       "      <td>80.0</td>\n",
       "      <td>670.0</td>\n",
       "      <td>A_16_P02204437</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NR_026743</td>\n",
       "      <td>tAKR</td>\n",
       "      <td>aldo-keto reductase, truncated</td>\n",
       "      <td>ref|NR_026743</td>\n",
       "      <td>chr10:4948382-4948441</td>\n",
       "      <td>hs|p15.1</td>\n",
       "      <td>Homo sapiens aldo-keto reductase, truncated (t...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>229614 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ID_REF     VALUE      ID    COL    ROW         SPOT_ID CONTROL_TYPE  \\\n",
       "235901  235902 -0.059363  235902  232.0  226.0  A_16_P41237987        FALSE   \n",
       "332477  332478  0.013735  332478  111.0  501.0    A_14_P101758        FALSE   \n",
       "184747  184748 -0.266349  184748  296.0  646.0  A_18_P10736965        FALSE   \n",
       "227995  227996 -0.134579  227996  242.0  118.0  A_16_P18922462        FALSE   \n",
       "242555  242556 -0.333080  242556  224.0  449.0  A_16_P18922623        FALSE   \n",
       "88687    88688 -0.193421   88688  417.0  134.0  A_16_P39030680        FALSE   \n",
       "242527  242528 -0.078122  242528  224.0  505.0  A_16_P02260125        FALSE   \n",
       "271992  271993  0.022711  271993  187.0  479.0  A_16_P02260134        FALSE   \n",
       "182310  182311 -0.148692  182311  299.0  744.0  A_16_P39030652        FALSE   \n",
       "53236    53237 -0.180240   53237  462.0  191.0  A_16_P39030621        FALSE   \n",
       "151091  151092 -0.134324  151092  339.0  297.0  A_16_P18922487        FALSE   \n",
       "97854    97855 -0.101267   97855  406.0  107.0  A_16_P18922549        FALSE   \n",
       "66678    66679 -0.382034   66679  445.0  371.0  A_16_P18922597        FALSE   \n",
       "28487    28488 -0.094625   28488  493.0  337.0  A_16_P32662933        FALSE   \n",
       "36946    36947 -0.166890   36947  482.0  136.0  A_16_P39030769        FALSE   \n",
       "382524  382525 -0.184075  382525   48.0  703.0  A_16_P18922640        FALSE   \n",
       "339939  339940 -0.093597  339940  101.0  702.0    A_14_P115717        FALSE   \n",
       "284504  284505 -0.051589  284505  171.0  132.0  A_16_P39030646        FALSE   \n",
       "174046  174047 -0.175303  174047  310.0  555.0    A_14_P127769        FALSE   \n",
       "49694    49695 -0.016057   49695  466.0  112.0  A_16_P03110179        FALSE   \n",
       "119756  119757  0.058620  119757  378.0   84.0  A_16_P20390559        FALSE   \n",
       "14971    14972 -0.131744   14972  510.0  305.0  A_16_P40547064        FALSE   \n",
       "306127  306128  0.071577  306128  144.0  665.0  A_18_P12377530        FALSE   \n",
       "340231  340232 -0.125075  340232  101.0  118.0  A_16_P20392765        FALSE   \n",
       "393200  393201  0.045828  393201   35.0   47.0  A_16_P03109077        FALSE   \n",
       "285713  285714 -0.028778  285714  170.0  101.0  A_18_P12380328        FALSE   \n",
       "63012    63013 -0.155494   63013  449.0  540.0  A_16_P20391759        FALSE   \n",
       "368379  368380  0.048246  368380   66.0  337.0  A_16_P03108492        FALSE   \n",
       "178940  178941  0.173739  178941  304.0  319.0  A_18_P12378393        FALSE   \n",
       "260875  260876  0.097303  260876  201.0  425.0  A_18_P12376163        FALSE   \n",
       "...        ...       ...     ...    ...    ...             ...          ...   \n",
       "304068  304069 -0.154090  304069  147.0    7.0  A_16_P15179920        FALSE   \n",
       "415770  415771  0.063581  415771    6.0  280.0    A_14_P106772        FALSE   \n",
       "124098  124099  0.077319  124099  373.0  155.0  A_16_P15179821        FALSE   \n",
       "331637  331638  0.237792  331638  112.0  589.0    A_14_P109546        FALSE   \n",
       "79674    79675  0.269498   79675  428.0  648.0  A_16_P15179878        FALSE   \n",
       "69060    69061  0.185846   69061  442.0  383.0    A_14_P101876        FALSE   \n",
       "281338  281339 -0.288873  281339  175.0   96.0    A_14_P125607        FALSE   \n",
       "47524    47525  0.130944   47525  469.0  471.0  A_16_P15179714        FALSE   \n",
       "267528  267529  0.055455  267529  192.0  652.0  A_16_P00101391        FALSE   \n",
       "175196  175197  0.173158  175197  308.0  644.0  A_16_P15179888        FALSE   \n",
       "54830    54831  0.152616   54831  460.0  187.0  A_16_P15179908        FALSE   \n",
       "6950      6951  0.140221    6951  520.0  427.0  A_18_P17656585        FALSE   \n",
       "249813  249814 -0.031044  249814  215.0  261.0  A_16_P35188503        FALSE   \n",
       "227335  227336 -0.053644  227336  243.0  641.0  A_16_P35188381        FALSE   \n",
       "51494    51495 -0.134389   51495  464.0  491.0  A_16_P15179845        FALSE   \n",
       "132936  132937 -0.075553  132937  361.0  788.0  A_16_P00101412        FALSE   \n",
       "89609    89610  0.037540   89610  416.0  677.0  A_16_P15179725        FALSE   \n",
       "163154  163155 -0.253419  163155  324.0   51.0  A_16_P34823706        FALSE   \n",
       "338273  338274 -0.117498  338274  104.0   53.0  A_16_P03598387        FALSE   \n",
       "96420    96421 -0.038295   96421  407.0  588.0  A_18_P13952367        FALSE   \n",
       "357092  357093 -0.054389  357093   80.0  623.0    A_14_P115613        FALSE   \n",
       "47466    47467 -0.046077   47467  469.0  587.0  A_16_P18818145        FALSE   \n",
       "107044  107045  0.062195  107045  394.0   36.0  A_16_P18818149        FALSE   \n",
       "103252  103253 -0.109646  103253  399.0  455.0  A_16_P18818159        FALSE   \n",
       "189784  189785 -0.025934  189785  290.0  124.0  A_16_P02204406        FALSE   \n",
       "399584  399585 -0.217337  399585   27.0   15.0  A_16_P02204415        FALSE   \n",
       "328696  328697 -0.059410  328697  116.0  103.0  A_16_P02204419        FALSE   \n",
       "269979  269980 -0.085281  269980  189.0  526.0  A_16_P02204427        FALSE   \n",
       "197213  197214  0.036307  197214  281.0  389.0  A_16_P02204398        FALSE   \n",
       "356671  356672 -0.084985  356672   80.0  670.0  A_16_P02204437        FALSE   \n",
       "\n",
       "              GB_ACC GENE_SYMBOL                          GENE_NAME  \\\n",
       "235901     NM_130786        A1BG             alpha-1-B glycoprotein   \n",
       "332477     NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "184747     NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "227995     NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "242555     NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "88687      NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "242527     NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "271992     NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "182310     NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "53236      NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "151091     NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "97854      NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "66678      NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "28487      NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "36946      NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "382524     NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "339939     NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "284504     NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "174046     NM_014576        A1CF     APOBEC1 complementation factor   \n",
       "49694   NM_001142333       A2BP1         ataxin 2-binding protein 1   \n",
       "119756  NM_001142333       A2BP1         ataxin 2-binding protein 1   \n",
       "14971   NM_001142333       A2BP1         ataxin 2-binding protein 1   \n",
       "306127  NM_001142333       A2BP1         ataxin 2-binding protein 1   \n",
       "340231  NM_001142333       A2BP1         ataxin 2-binding protein 1   \n",
       "393200  NM_001142333       A2BP1         ataxin 2-binding protein 1   \n",
       "285713  NM_001142333       A2BP1         ataxin 2-binding protein 1   \n",
       "63012   NM_001142333       A2BP1         ataxin 2-binding protein 1   \n",
       "368379  NM_001142333       A2BP1         ataxin 2-binding protein 1   \n",
       "178940  NM_001142333       A2BP1         ataxin 2-binding protein 1   \n",
       "260875  NM_001142333       A2BP1         ataxin 2-binding protein 1   \n",
       "...              ...         ...                                ...   \n",
       "304068     NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "415770     NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "124098     NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "331637     NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "79674      NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "69060      NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "281338     NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "47524      NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "267528     NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "175196     NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "54830      NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "6950       NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "249813     NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "227335     NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "51494      NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "132936     NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "89609      NM_015534        ZZZ3  zinc finger, ZZ-type containing 3   \n",
       "163154     NR_001591   psiTPTE22                    TPTE pseudogene   \n",
       "338273     NR_001591   psiTPTE22                    TPTE pseudogene   \n",
       "96420      NR_001591   psiTPTE22                    TPTE pseudogene   \n",
       "357092     NR_001591   psiTPTE22                    TPTE pseudogene   \n",
       "47466      NR_026743        tAKR     aldo-keto reductase, truncated   \n",
       "107044     NR_026743        tAKR     aldo-keto reductase, truncated   \n",
       "103252     NR_026743        tAKR     aldo-keto reductase, truncated   \n",
       "189784     NR_026743        tAKR     aldo-keto reductase, truncated   \n",
       "399584     NR_026743        tAKR     aldo-keto reductase, truncated   \n",
       "328696     NR_026743        tAKR     aldo-keto reductase, truncated   \n",
       "269979     NR_026743        tAKR     aldo-keto reductase, truncated   \n",
       "197213     NR_026743        tAKR     aldo-keto reductase, truncated   \n",
       "356671     NR_026743        tAKR     aldo-keto reductase, truncated   \n",
       "\n",
       "                                         ACCESSION_STRING  \\\n",
       "235901                                      ref|NM_130786   \n",
       "332477          ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "184747          ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "227995          ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "242555          ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "88687           ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "242527          ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "271992          ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "182310          ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "53236           ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "151091          ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "97854           ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "66678           ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "28487           ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "36946           ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "382524          ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "339939          ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "284504          ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "174046          ref|NM_014576|ref|NM_138932|ref|NM_138933   \n",
       "49694   ref|NM_001142333|ref|NM_018723|ref|NM_00114233...   \n",
       "119756                     ref|NM_001142333|ref|NM_018723   \n",
       "14971                      ref|NM_001142333|ref|NM_018723   \n",
       "306127                     ref|NM_001142333|ref|NM_018723   \n",
       "340231  ref|NM_001142333|ref|NM_018723|ref|NM_00114233...   \n",
       "393200                     ref|NM_001142333|ref|NM_018723   \n",
       "285713    ref|NM_001142333|ref|NM_018723|ref|NM_001142334   \n",
       "63012     ref|NM_001142333|ref|NM_018723|ref|NM_001142334   \n",
       "368379                     ref|NM_001142333|ref|NM_018723   \n",
       "178940                     ref|NM_001142333|ref|NM_018723   \n",
       "260875                     ref|NM_001142333|ref|NM_018723   \n",
       "...                                                   ...   \n",
       "304068                                      ref|NM_015534   \n",
       "415770                                      ref|NM_015534   \n",
       "124098                                      ref|NM_015534   \n",
       "331637                                      ref|NM_015534   \n",
       "79674                                       ref|NM_015534   \n",
       "69060                                       ref|NM_015534   \n",
       "281338                                      ref|NM_015534   \n",
       "47524                                       ref|NM_015534   \n",
       "267528                                      ref|NM_015534   \n",
       "175196                                      ref|NM_015534   \n",
       "54830                                       ref|NM_015534   \n",
       "6950                                        ref|NM_015534   \n",
       "249813                                      ref|NM_015534   \n",
       "227335                                      ref|NM_015534   \n",
       "51494                                       ref|NM_015534   \n",
       "132936                                      ref|NM_015534   \n",
       "89609                                       ref|NM_015534   \n",
       "163154                                      ref|NR_001591   \n",
       "338273                                      ref|NR_001591   \n",
       "96420                                       ref|NR_001591   \n",
       "357092                                      ref|NR_001591   \n",
       "47466                                       ref|NR_026743   \n",
       "107044                                      ref|NR_026743   \n",
       "103252                                      ref|NR_026743   \n",
       "189784                                      ref|NR_026743   \n",
       "399584                                      ref|NR_026743   \n",
       "328696                                      ref|NR_026743   \n",
       "269979                                      ref|NR_026743   \n",
       "197213                                      ref|NR_026743   \n",
       "356671                                      ref|NR_026743   \n",
       "\n",
       "           CHROMOSOMAL_LOCATION   CYTOBAND  \\\n",
       "235901  chr19:63551904-63551963  hs|q13.43   \n",
       "332477  chr10:52315385-52315441  hs|q11.23   \n",
       "184747  chr10:52310289-52310348  hs|q11.23   \n",
       "227995  chr10:52245714-52245773  hs|q11.23   \n",
       "242555  chr10:52301417-52301476  hs|q11.23   \n",
       "88687   chr10:52271703-52271762  hs|q11.23   \n",
       "242527  chr10:52285879-52285938  hs|q11.23   \n",
       "271992  chr10:52289698-52289757  hs|q11.23   \n",
       "182310  chr10:52262400-52262459  hs|q11.23   \n",
       "53236   chr10:52249696-52249755  hs|q11.23   \n",
       "151091  chr10:52252979-52253038  hs|q11.23   \n",
       "97854   chr10:52278706-52278765  hs|q11.23   \n",
       "66678   chr10:52292685-52292744  hs|q11.23   \n",
       "28487   chr10:52240955-52241014  hs|q11.23   \n",
       "36946   chr10:52298312-52298371  hs|q11.23   \n",
       "382524  chr10:52306765-52306824  hs|q11.23   \n",
       "339939  chr10:52265960-52266019  hs|q11.23   \n",
       "284504  chr10:52258030-52258089  hs|q11.23   \n",
       "174046  chr10:52236360-52236419  hs|q11.23   \n",
       "49694     chr16:7387703-7387762   hs|p13.2   \n",
       "119756    chr16:6634165-6634224   hs|p13.2   \n",
       "14971     chr16:6609413-6609472   hs|p13.2   \n",
       "306127    chr16:6159042-6159101   hs|p13.3   \n",
       "340231    chr16:7449772-7449831   hs|p13.2   \n",
       "393200    chr16:6719675-6719734   hs|p13.2   \n",
       "285713    chr16:7166984-7167043   hs|p13.2   \n",
       "63012     chr16:7101613-7101672   hs|p13.2   \n",
       "368379    chr16:6364057-6364116   hs|p13.2   \n",
       "178940    chr16:6647650-6647709   hs|p13.2   \n",
       "260875    chr16:6198892-6198951   hs|p13.3   \n",
       "...                         ...        ...   \n",
       "304068   chr1:77899238-77899297   hs|p31.1   \n",
       "415770   chr1:77820120-77820179   hs|p31.1   \n",
       "124098   chr1:77857751-77857810   hs|p31.1   \n",
       "331637   chr1:77819229-77819288   hs|p31.1   \n",
       "79674    chr1:77876492-77876551   hs|p31.1   \n",
       "69060    chr1:77903687-77903746   hs|p31.1   \n",
       "281338   chr1:77871494-77871553   hs|p31.1   \n",
       "47524    chr1:77807227-77807286   hs|p31.1   \n",
       "267528   chr1:77884802-77884861   hs|p31.1   \n",
       "175196   chr1:77879708-77879767   hs|p31.1   \n",
       "54830    chr1:77894759-77894818   hs|p31.1   \n",
       "6950     chr1:77829907-77829966   hs|p31.1   \n",
       "249813   chr1:77911289-77911348   hs|p31.1   \n",
       "227335   chr1:77860965-77861024   hs|p31.1   \n",
       "51494    chr1:77866835-77866894   hs|p31.1   \n",
       "132936   chr1:77908354-77908413   hs|p31.1   \n",
       "89609    chr1:77812118-77812177   hs|p31.1   \n",
       "163154  chr22:15508442-15508499   hs|q11.1   \n",
       "338273  chr22:15464955-15465014   hs|q11.1   \n",
       "96420   chr22:15471221-15471273   hs|q11.1   \n",
       "357092  chr22:15476855-15476914   hs|q11.1   \n",
       "47466     chr10:4934320-4934379   hs|p15.1   \n",
       "107044    chr10:4939503-4939562   hs|p15.1   \n",
       "103252    chr10:4945256-4945315   hs|p15.1   \n",
       "189784    chr10:4909104-4909163   hs|p15.1   \n",
       "399584    chr10:4918272-4918331   hs|p15.1   \n",
       "328696    chr10:4920849-4920908   hs|p15.1   \n",
       "269979    chr10:4924101-4924160   hs|p15.1   \n",
       "197213    chr10:4904018-4904077   hs|p15.1   \n",
       "356671    chr10:4948382-4948441   hs|p15.1   \n",
       "\n",
       "                                              DESCRIPTION  \n",
       "235901  Homo sapiens alpha-1-B glycoprotein (A1BG), mRNA.  \n",
       "332477  Homo sapiens APOBEC1 complementation factor (A...  \n",
       "184747  Homo sapiens APOBEC1 complementation factor (A...  \n",
       "227995  Homo sapiens APOBEC1 complementation factor (A...  \n",
       "242555  Homo sapiens APOBEC1 complementation factor (A...  \n",
       "88687   Homo sapiens APOBEC1 complementation factor (A...  \n",
       "242527  Homo sapiens APOBEC1 complementation factor (A...  \n",
       "271992  Homo sapiens APOBEC1 complementation factor (A...  \n",
       "182310  Homo sapiens APOBEC1 complementation factor (A...  \n",
       "53236   Homo sapiens APOBEC1 complementation factor (A...  \n",
       "151091  Homo sapiens APOBEC1 complementation factor (A...  \n",
       "97854   Homo sapiens APOBEC1 complementation factor (A...  \n",
       "66678   Homo sapiens APOBEC1 complementation factor (A...  \n",
       "28487   Homo sapiens APOBEC1 complementation factor (A...  \n",
       "36946   Homo sapiens APOBEC1 complementation factor (A...  \n",
       "382524  Homo sapiens APOBEC1 complementation factor (A...  \n",
       "339939  Homo sapiens APOBEC1 complementation factor (A...  \n",
       "284504  Homo sapiens APOBEC1 complementation factor (A...  \n",
       "174046  Homo sapiens APOBEC1 complementation factor (A...  \n",
       "49694   Homo sapiens ataxin 2-binding protein 1 (A2BP1...  \n",
       "119756  Homo sapiens ataxin 2-binding protein 1 (A2BP1...  \n",
       "14971   Homo sapiens ataxin 2-binding protein 1 (A2BP1...  \n",
       "306127  Homo sapiens ataxin 2-binding protein 1 (A2BP1...  \n",
       "340231  Homo sapiens ataxin 2-binding protein 1 (A2BP1...  \n",
       "393200  Homo sapiens ataxin 2-binding protein 1 (A2BP1...  \n",
       "285713  Homo sapiens ataxin 2-binding protein 1 (A2BP1...  \n",
       "63012   Homo sapiens ataxin 2-binding protein 1 (A2BP1...  \n",
       "368379  Homo sapiens ataxin 2-binding protein 1 (A2BP1...  \n",
       "178940  Homo sapiens ataxin 2-binding protein 1 (A2BP1...  \n",
       "260875  Homo sapiens ataxin 2-binding protein 1 (A2BP1...  \n",
       "...                                                   ...  \n",
       "304068  Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "415770  Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "124098  Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "331637  Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "79674   Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "69060   Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "281338  Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "47524   Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "267528  Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "175196  Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "54830   Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "6950    Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "249813  Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "227335  Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "51494   Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "132936  Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "89609   Homo sapiens zinc finger, ZZ-type containing 3...  \n",
       "163154  Homo sapiens TPTE pseudogene (psiTPTE22), non-...  \n",
       "338273  Homo sapiens TPTE pseudogene (psiTPTE22), non-...  \n",
       "96420   Homo sapiens TPTE pseudogene (psiTPTE22), non-...  \n",
       "357092  Homo sapiens TPTE pseudogene (psiTPTE22), non-...  \n",
       "47466   Homo sapiens aldo-keto reductase, truncated (t...  \n",
       "107044  Homo sapiens aldo-keto reductase, truncated (t...  \n",
       "103252  Homo sapiens aldo-keto reductase, truncated (t...  \n",
       "189784  Homo sapiens aldo-keto reductase, truncated (t...  \n",
       "399584  Homo sapiens aldo-keto reductase, truncated (t...  \n",
       "328696  Homo sapiens aldo-keto reductase, truncated (t...  \n",
       "269979  Homo sapiens aldo-keto reductase, truncated (t...  \n",
       "197213  Homo sapiens aldo-keto reductase, truncated (t...  \n",
       "356671  Homo sapiens aldo-keto reductase, truncated (t...  \n",
       "\n",
       "[229614 rows x 14 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Easy way to verify that GENE_SYMBOL is not unique in the CNV data: \n",
    "# just sort and see if we see multiple rows with the same gene. We do.\n",
    "cnv_ratios_and_genenames.sort_values(\"GENE_SYMBOL\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We really want to reduce these data to a single median value per gene, as genes are the atomic unit in our sgRNA data. (\"VALUE\" is the column containing the log10 ratios of control to A375 spot intensity). log10 Cy5/Cy3, i.e. A375/control. Greater than 0 is enriched in A375; less than 1 is depleted in A375."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"3\" halign=\"left\">VALUE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GENE_SYMBOL</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.059363</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>-0.141636</td>\n",
       "      <td>0.104701</td>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2BP1</th>\n",
       "      <td>-0.005361</td>\n",
       "      <td>0.156898</td>\n",
       "      <td>343.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2LD1</th>\n",
       "      <td>-0.091086</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.016620</td>\n",
       "      <td>0.117057</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                VALUE                 \n",
       "               median       std    len\n",
       "GENE_SYMBOL                           \n",
       "A1BG        -0.059363       NaN    1.0\n",
       "A1CF        -0.141636  0.104701   18.0\n",
       "A2BP1       -0.005361  0.156898  343.0\n",
       "A2LD1       -0.091086       NaN    1.0\n",
       "A2M         -0.016620  0.117057    7.0"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv_ratios_and_genenames[[\"VALUE\", \"GENE_SYMBOL\"]]\n",
    "cnv_grouped_by_gene = cnv_ratios_and_genenames[[\"VALUE\", \"GENE_SYMBOL\"]].groupby('GENE_SYMBOL')\n",
    "cnv_summary_by_gene = cnv_grouped_by_gene.agg([numpy.median, numpy.std, len])\n",
    "cnv_summary_by_gene.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We need to do something similar for the sgRNA data. However, this is a little more complicated as we first need to normalize the cell data against the starting plasmid concentration. The normalized plasmid abundances are 1-based, so simply dividing the sequenced cell-guide abundance by the plasmid input data should be sufficient."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.0"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "min(adata[\"norm_count_plasmid\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>spacer_id</th>\n",
       "      <th>spacer_seq</th>\n",
       "      <th>norm_count_plasmid</th>\n",
       "      <th>norm_count_D7_Rep1</th>\n",
       "      <th>norm_count_D7_Rep2</th>\n",
       "      <th>norm_count_D14_Rep1</th>\n",
       "      <th>norm_count_D14_Rep2</th>\n",
       "      <th>norm_count_PLX7_Rep1</th>\n",
       "      <th>norm_count_PLX7_Rep2</th>\n",
       "      <th>norm_count_PLX14_Rep1</th>\n",
       "      <th>norm_count_PLX14_Rep2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>s_1</td>\n",
       "      <td>ACCTGTAGTTGCCGGCGTGC</td>\n",
       "      <td>10.366767</td>\n",
       "      <td>8.142715</td>\n",
       "      <td>10.177020</td>\n",
       "      <td>7.986137</td>\n",
       "      <td>6.461769</td>\n",
       "      <td>12.260794</td>\n",
       "      <td>9.095489</td>\n",
       "      <td>4.285777</td>\n",
       "      <td>2.816931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>s_2</td>\n",
       "      <td>GTCGCTGAGCTCCGATTCGA</td>\n",
       "      <td>8.805639</td>\n",
       "      <td>9.941240</td>\n",
       "      <td>8.059246</td>\n",
       "      <td>8.605162</td>\n",
       "      <td>6.461769</td>\n",
       "      <td>7.491095</td>\n",
       "      <td>10.515751</td>\n",
       "      <td>4.168428</td>\n",
       "      <td>5.239506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>s_3</td>\n",
       "      <td>AAGTTGCTTGATTGCATTCT</td>\n",
       "      <td>19.493360</td>\n",
       "      <td>26.025195</td>\n",
       "      <td>25.203128</td>\n",
       "      <td>33.277722</td>\n",
       "      <td>21.514938</td>\n",
       "      <td>19.253963</td>\n",
       "      <td>27.913951</td>\n",
       "      <td>9.801189</td>\n",
       "      <td>9.327602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>s_4</td>\n",
       "      <td>ACTTCCCCGAGACCTTTTTG</td>\n",
       "      <td>20.454054</td>\n",
       "      <td>23.147554</td>\n",
       "      <td>23.387894</td>\n",
       "      <td>28.856116</td>\n",
       "      <td>20.649048</td>\n",
       "      <td>20.688459</td>\n",
       "      <td>21.167710</td>\n",
       "      <td>7.219507</td>\n",
       "      <td>8.822898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>s_5</td>\n",
       "      <td>AGTGCCGCTCAATCCATCCC</td>\n",
       "      <td>40.868803</td>\n",
       "      <td>54.801599</td>\n",
       "      <td>54.952807</td>\n",
       "      <td>56.270072</td>\n",
       "      <td>53.353057</td>\n",
       "      <td>51.996336</td>\n",
       "      <td>47.868622</td>\n",
       "      <td>12.206848</td>\n",
       "      <td>13.768989</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_name spacer_id            spacer_seq  norm_count_plasmid  \\\n",
       "0      A1BG       s_1  ACCTGTAGTTGCCGGCGTGC           10.366767   \n",
       "1      A1BG       s_2  GTCGCTGAGCTCCGATTCGA            8.805639   \n",
       "2      A1CF       s_3  AAGTTGCTTGATTGCATTCT           19.493360   \n",
       "3      A1CF       s_4  ACTTCCCCGAGACCTTTTTG           20.454054   \n",
       "4      A1CF       s_5  AGTGCCGCTCAATCCATCCC           40.868803   \n",
       "\n",
       "   norm_count_D7_Rep1  norm_count_D7_Rep2  norm_count_D14_Rep1  \\\n",
       "0            8.142715           10.177020             7.986137   \n",
       "1            9.941240            8.059246             8.605162   \n",
       "2           26.025195           25.203128            33.277722   \n",
       "3           23.147554           23.387894            28.856116   \n",
       "4           54.801599           54.952807            56.270072   \n",
       "\n",
       "   norm_count_D14_Rep2  norm_count_PLX7_Rep1  norm_count_PLX7_Rep2  \\\n",
       "0             6.461769             12.260794              9.095489   \n",
       "1             6.461769              7.491095             10.515751   \n",
       "2            21.514938             19.253963             27.913951   \n",
       "3            20.649048             20.688459             21.167710   \n",
       "4            53.353057             51.996336             47.868622   \n",
       "\n",
       "   norm_count_PLX14_Rep1  norm_count_PLX14_Rep2  \n",
       "0               4.285777               2.816931  \n",
       "1               4.168428               5.239506  \n",
       "2               9.801189               9.327602  \n",
       "3               7.219507               8.822898  \n",
       "4              12.206848              13.768989  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "guide_abundance = adata.iloc[:,4:].div(adata[\"norm_count_plasmid\"], axis='index')\n",
    "guide_abundance = guide_abundance.applymap(numpy.log)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Join the normalized data back with the guide/gene information (this function matches on the row index, avoiding off-by-one errors)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Merges the first three columns of the original data with the newly normalized data\n",
    "guide_abundance = pandas.merge(adata.iloc[:,:3], guide_abundance, left_index=True, right_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "grouped = guide_abundance.groupby('gene_name')\n",
    "guide_abundance = grouped.agg([numpy.median, numpy.std, len])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_D7_Rep1</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_D7_Rep2</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_D14_Rep1</th>\n",
       "      <th>norm_count_D14_Rep2</th>\n",
       "      <th>...</th>\n",
       "      <th>norm_count_PLX7_Rep1</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_PLX7_Rep2</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_PLX14_Rep1</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_PLX14_Rep2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>...</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>0.190696</td>\n",
       "      <td>0.281610</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.045635</td>\n",
       "      <td>0.283213</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.393056</td>\n",
       "      <td>0.350848</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.012581</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.010728</td>\n",
       "      <td>0.397092</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.830710</td>\n",
       "      <td>0.540743</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.708580</td>\n",
       "      <td>0.888194</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>-0.175949</td>\n",
       "      <td>0.283158</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.128629</td>\n",
       "      <td>0.185996</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.458150</td>\n",
       "      <td>0.149402</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.103529</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.147201</td>\n",
       "      <td>0.406511</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.219735</td>\n",
       "      <td>0.768044</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.404130</td>\n",
       "      <td>0.949658</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>-0.344126</td>\n",
       "      <td>0.429897</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.412194</td>\n",
       "      <td>0.478731</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.590130</td>\n",
       "      <td>0.480176</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.231557</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.440484</td>\n",
       "      <td>0.570848</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.022066</td>\n",
       "      <td>0.540522</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.151246</td>\n",
       "      <td>0.544604</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>-0.631462</td>\n",
       "      <td>0.592196</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.288968</td>\n",
       "      <td>0.464338</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.780016</td>\n",
       "      <td>0.637097</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.206432</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.428154</td>\n",
       "      <td>0.508536</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-1.473201</td>\n",
       "      <td>0.518845</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-1.423504</td>\n",
       "      <td>0.548200</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hsa-mir-3187</th>\n",
       "      <td>-0.283218</td>\n",
       "      <td>0.430932</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.162773</td>\n",
       "      <td>0.451196</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.733347</td>\n",
       "      <td>0.406174</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.097553</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.252364</td>\n",
       "      <td>0.402374</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.753625</td>\n",
       "      <td>0.273931</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.477145</td>\n",
       "      <td>0.096784</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             norm_count_D7_Rep1                norm_count_D7_Rep2            \\\n",
       "                         median       std  len             median       std   \n",
       "gene_name                                                                     \n",
       "ZYG11B                 0.190696  0.281610  3.0           0.045635  0.283213   \n",
       "ZYX                   -0.175949  0.283158  4.0          -0.128629  0.185996   \n",
       "ZZEF1                 -0.344126  0.429897  4.0          -0.412194  0.478731   \n",
       "ZZZ3                  -0.631462  0.592196  6.0          -0.288968  0.464338   \n",
       "hsa-mir-3187          -0.283218  0.430932  3.0          -0.162773  0.451196   \n",
       "\n",
       "                  norm_count_D14_Rep1                norm_count_D14_Rep2 ...   \\\n",
       "              len              median       std  len              median ...    \n",
       "gene_name                                                                ...    \n",
       "ZYG11B        3.0            0.393056  0.350848  3.0            0.012581 ...    \n",
       "ZYX           4.0           -0.458150  0.149402  4.0           -0.103529 ...    \n",
       "ZZEF1         4.0           -0.590130  0.480176  4.0           -0.231557 ...    \n",
       "ZZZ3          6.0           -0.780016  0.637097  6.0           -0.206432 ...    \n",
       "hsa-mir-3187  3.0           -0.733347  0.406174  3.0           -0.097553 ...    \n",
       "\n",
       "             norm_count_PLX7_Rep1 norm_count_PLX7_Rep2                 \\\n",
       "                              len               median       std  len   \n",
       "gene_name                                                               \n",
       "ZYG11B                        3.0             0.010728  0.397092  3.0   \n",
       "ZYX                           4.0            -0.147201  0.406511  4.0   \n",
       "ZZEF1                         4.0            -0.440484  0.570848  4.0   \n",
       "ZZZ3                          6.0            -0.428154  0.508536  6.0   \n",
       "hsa-mir-3187                  3.0            -0.252364  0.402374  3.0   \n",
       "\n",
       "             norm_count_PLX14_Rep1                norm_count_PLX14_Rep2  \\\n",
       "                            median       std  len                median   \n",
       "gene_name                                                                 \n",
       "ZYG11B                   -0.830710  0.540743  3.0             -0.708580   \n",
       "ZYX                      -1.219735  0.768044  4.0             -1.404130   \n",
       "ZZEF1                    -1.022066  0.540522  4.0             -1.151246   \n",
       "ZZZ3                     -1.473201  0.518845  6.0             -1.423504   \n",
       "hsa-mir-3187             -1.753625  0.273931  3.0             -1.477145   \n",
       "\n",
       "                             \n",
       "                   std  len  \n",
       "gene_name                    \n",
       "ZYG11B        0.888194  3.0  \n",
       "ZYX           0.949658  4.0  \n",
       "ZZEF1         0.544604  4.0  \n",
       "ZZZ3          0.548200  6.0  \n",
       "hsa-mir-3187  0.096784  3.0  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guide_abundance.tail()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next concern is how well the datasets map onto each other. Are there many genes targeted by guides that for some reason aren't in the CNV data, or vice-versa?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17808\n"
     ]
    }
   ],
   "source": [
    "print(len(cnv_summary_by_gene))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17419\n"
     ]
    }
   ],
   "source": [
    "print(len(guide_abundance))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So, there are 17808 genes in the CNV dataset, but only 17419 in the sgRNA dataset. That's not unexpected; some sgRNAs are likely to be totally lost in library preparation, whereas all intended spots are imaged on the CNV array."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15268\n",
      "19959\n"
     ]
    }
   ],
   "source": [
    "# First the inner join (selects for genes in both datasets):\n",
    "cnv_and_guides = pandas.merge(cnv_summary_by_gene, guide_abundance, left_index=True, right_index=True, how=\"inner\")\n",
    "print(len(cnv_and_guides))\n",
    "\n",
    "# Next the outer join (selects for genes in either dataset):\n",
    "cnv_and_guides = pandas.merge(cnv_summary_by_gene, guide_abundance, left_index=True, right_index=True, how=\"outer\")\n",
    "print(len(cnv_and_guides))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Rename some columns to help keep track.\n",
    "cnv_and_guides =cnv_and_guides.rename(columns={'VALUE': 'CNV'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are 19,959 genes when you pool both datasets, but we only have both sgRNA abundance data and CNV data for 15,268 of them. Ultimately we don't have a choice but to work only with those 15,268, but it's worth seeing what the non-matching genes are to check if we've done something wrong."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"3\" halign=\"left\">CNV</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_D7_Rep1</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_D7_Rep2</th>\n",
       "      <th>norm_count_D14_Rep1</th>\n",
       "      <th>...</th>\n",
       "      <th>norm_count_PLX7_Rep1</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_PLX7_Rep2</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_PLX14_Rep1</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_PLX14_Rep2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>...</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1-Dec</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.096822</td>\n",
       "      <td>0.309539</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.313602</td>\n",
       "      <td>0.252063</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.156710</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.142697</td>\n",
       "      <td>0.429846</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.738297</td>\n",
       "      <td>0.488660</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.649533</td>\n",
       "      <td>0.582200</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.001451</td>\n",
       "      <td>0.149900</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-0.007358</td>\n",
       "      <td>0.238821</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.144568</td>\n",
       "      <td>...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-0.160655</td>\n",
       "      <td>0.428572</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-1.105906</td>\n",
       "      <td>0.597998</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-1.053001</td>\n",
       "      <td>0.656827</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.250745</td>\n",
       "      <td>0.104664</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.154311</td>\n",
       "      <td>0.324163</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.389285</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.164021</td>\n",
       "      <td>0.452095</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.263459</td>\n",
       "      <td>0.374423</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.215216</td>\n",
       "      <td>0.361984</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.316717</td>\n",
       "      <td>0.359520</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.290522</td>\n",
       "      <td>0.378278</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.143517</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.014089</td>\n",
       "      <td>0.283933</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.155326</td>\n",
       "      <td>0.484769</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.987062</td>\n",
       "      <td>0.601717</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.541840</td>\n",
       "      <td>0.514727</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.460078</td>\n",
       "      <td>0.578075</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.547001</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.197123</td>\n",
       "      <td>0.467431</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.800630</td>\n",
       "      <td>0.352074</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.109213</td>\n",
       "      <td>0.546510</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          CNV         norm_count_D7_Rep1                norm_count_D7_Rep2  \\\n",
       "       median std len             median       std  len             median   \n",
       "1-Dec     NaN NaN NaN          -0.096822  0.309539  3.0          -0.313602   \n",
       "1-Mar     NaN NaN NaN          -0.001451  0.149900  9.0          -0.007358   \n",
       "1-Sep     NaN NaN NaN          -0.250745  0.104664  3.0          -0.154311   \n",
       "10-Mar    NaN NaN NaN           0.316717  0.359520  4.0           0.290522   \n",
       "10-Sep    NaN NaN NaN          -0.541840  0.514727  3.0          -0.460078   \n",
       "\n",
       "                      norm_count_D14_Rep1 ...  norm_count_PLX7_Rep1  \\\n",
       "             std  len              median ...                   len   \n",
       "1-Dec   0.252063  3.0           -0.156710 ...                   3.0   \n",
       "1-Mar   0.238821  9.0            0.144568 ...                   9.0   \n",
       "1-Sep   0.324163  3.0           -0.389285 ...                   3.0   \n",
       "10-Mar  0.378278  4.0            0.143517 ...                   4.0   \n",
       "10-Sep  0.578075  3.0           -0.547001 ...                   3.0   \n",
       "\n",
       "       norm_count_PLX7_Rep2                norm_count_PLX14_Rep1            \\\n",
       "                     median       std  len                median       std   \n",
       "1-Dec             -0.142697  0.429846  3.0             -0.738297  0.488660   \n",
       "1-Mar             -0.160655  0.428572  9.0             -1.105906  0.597998   \n",
       "1-Sep             -0.164021  0.452095  3.0             -1.263459  0.374423   \n",
       "10-Mar            -0.014089  0.283933  4.0             -1.155326  0.484769   \n",
       "10-Sep            -0.197123  0.467431  3.0             -0.800630  0.352074   \n",
       "\n",
       "            norm_count_PLX14_Rep2                 \n",
       "        len                median       std  len  \n",
       "1-Dec   3.0             -0.649533  0.582200  3.0  \n",
       "1-Mar   9.0             -1.053001  0.656827  9.0  \n",
       "1-Sep   3.0             -1.215216  0.361984  3.0  \n",
       "10-Mar  4.0             -0.987062  0.601717  4.0  \n",
       "10-Sep  3.0             -1.109213  0.546510  3.0  \n",
       "\n",
       "[5 rows x 27 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv_and_guides.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"3\" halign=\"left\">CNV</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_D7_Rep1</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_D7_Rep2</th>\n",
       "      <th>norm_count_D14_Rep1</th>\n",
       "      <th>...</th>\n",
       "      <th>norm_count_PLX7_Rep1</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_PLX7_Rep2</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_PLX14_Rep1</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_PLX14_Rep2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>...</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1-Dec</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.096822</td>\n",
       "      <td>0.309539</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.313602</td>\n",
       "      <td>0.252063</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.156710</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.142697</td>\n",
       "      <td>0.429846</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.738297</td>\n",
       "      <td>0.488660</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.649533</td>\n",
       "      <td>0.582200</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.001451</td>\n",
       "      <td>0.149900</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-0.007358</td>\n",
       "      <td>0.238821</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.144568</td>\n",
       "      <td>...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-0.160655</td>\n",
       "      <td>0.428572</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-1.105906</td>\n",
       "      <td>0.597998</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-1.053001</td>\n",
       "      <td>0.656827</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.250745</td>\n",
       "      <td>0.104664</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.154311</td>\n",
       "      <td>0.324163</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.389285</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.164021</td>\n",
       "      <td>0.452095</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.263459</td>\n",
       "      <td>0.374423</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.215216</td>\n",
       "      <td>0.361984</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.316717</td>\n",
       "      <td>0.359520</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.290522</td>\n",
       "      <td>0.378278</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.143517</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.014089</td>\n",
       "      <td>0.283933</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.155326</td>\n",
       "      <td>0.484769</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.987062</td>\n",
       "      <td>0.601717</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.541840</td>\n",
       "      <td>0.514727</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.460078</td>\n",
       "      <td>0.578075</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.547001</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.197123</td>\n",
       "      <td>0.467431</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.800630</td>\n",
       "      <td>0.352074</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.109213</td>\n",
       "      <td>0.546510</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.120893</td>\n",
       "      <td>0.221079</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.230257</td>\n",
       "      <td>0.336709</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.159451</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.101848</td>\n",
       "      <td>0.155371</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.509583</td>\n",
       "      <td>0.633365</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.215090</td>\n",
       "      <td>0.983348</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.133383</td>\n",
       "      <td>0.116819</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.077200</td>\n",
       "      <td>0.065924</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.331909</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.076066</td>\n",
       "      <td>0.004554</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.932400</td>\n",
       "      <td>0.274641</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.244964</td>\n",
       "      <td>0.272671</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005727</td>\n",
       "      <td>0.072789</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.032687</td>\n",
       "      <td>0.133818</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.037297</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.039585</td>\n",
       "      <td>0.109875</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.450022</td>\n",
       "      <td>0.122485</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.377543</td>\n",
       "      <td>0.146387</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.290554</td>\n",
       "      <td>0.306691</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.142270</td>\n",
       "      <td>0.413997</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.089395</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.594866</td>\n",
       "      <td>0.327882</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.547629</td>\n",
       "      <td>0.321109</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.709218</td>\n",
       "      <td>0.409802</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.242636</td>\n",
       "      <td>0.223773</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.052407</td>\n",
       "      <td>0.250299</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.300373</td>\n",
       "      <td>...</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.115797</td>\n",
       "      <td>0.335042</td>\n",
       "      <td>11.0</td>\n",
       "      <td>-0.788051</td>\n",
       "      <td>0.716257</td>\n",
       "      <td>11.0</td>\n",
       "      <td>-0.852748</td>\n",
       "      <td>0.710373</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.074227</td>\n",
       "      <td>0.149550</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.067209</td>\n",
       "      <td>0.195979</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.077875</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.071584</td>\n",
       "      <td>0.121537</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.473724</td>\n",
       "      <td>0.705837</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.435851</td>\n",
       "      <td>0.480136</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.077972</td>\n",
       "      <td>0.152052</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.130657</td>\n",
       "      <td>0.307455</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.022833</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.116003</td>\n",
       "      <td>0.218735</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.742879</td>\n",
       "      <td>0.584929</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.685291</td>\n",
       "      <td>0.749166</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.046410</td>\n",
       "      <td>0.156341</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.008428</td>\n",
       "      <td>0.260490</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.093611</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.008398</td>\n",
       "      <td>0.267759</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.007701</td>\n",
       "      <td>0.547351</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.218126</td>\n",
       "      <td>0.494752</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.200761</td>\n",
       "      <td>0.241437</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.188366</td>\n",
       "      <td>0.138867</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-0.220544</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.150098</td>\n",
       "      <td>0.290600</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-0.932872</td>\n",
       "      <td>0.731989</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-1.350516</td>\n",
       "      <td>0.870646</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.073473</td>\n",
       "      <td>0.370888</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.071134</td>\n",
       "      <td>0.171446</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.021822</td>\n",
       "      <td>...</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.019595</td>\n",
       "      <td>0.317571</td>\n",
       "      <td>8.0</td>\n",
       "      <td>-0.877557</td>\n",
       "      <td>0.502653</td>\n",
       "      <td>8.0</td>\n",
       "      <td>-0.840931</td>\n",
       "      <td>0.603345</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.364824</td>\n",
       "      <td>0.234827</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.270827</td>\n",
       "      <td>0.102993</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.180895</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.107403</td>\n",
       "      <td>0.455855</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.999522</td>\n",
       "      <td>1.551640</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.627167</td>\n",
       "      <td>1.267703</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.204853</td>\n",
       "      <td>0.070917</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-0.166603</td>\n",
       "      <td>0.166208</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-0.363972</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-0.139331</td>\n",
       "      <td>0.124380</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-0.927778</td>\n",
       "      <td>0.431610</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-1.396363</td>\n",
       "      <td>0.408007</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.080671</td>\n",
       "      <td>0.646677</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.081839</td>\n",
       "      <td>0.491706</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.095961</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.334222</td>\n",
       "      <td>0.419352</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.136917</td>\n",
       "      <td>0.241243</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.248705</td>\n",
       "      <td>0.274294</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.316024</td>\n",
       "      <td>0.114487</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.208226</td>\n",
       "      <td>0.163103</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.503353</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.252912</td>\n",
       "      <td>0.116103</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.490197</td>\n",
       "      <td>0.334553</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.409739</td>\n",
       "      <td>0.386353</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.103194</td>\n",
       "      <td>0.370343</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.141245</td>\n",
       "      <td>0.247514</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.176513</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.219990</td>\n",
       "      <td>0.269957</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.953055</td>\n",
       "      <td>0.356929</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.962991</td>\n",
       "      <td>0.292234</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.043177</td>\n",
       "      <td>0.139767</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.022048</td>\n",
       "      <td>0.033383</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.044235</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.002691</td>\n",
       "      <td>0.121394</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.016384</td>\n",
       "      <td>0.570748</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.066330</td>\n",
       "      <td>0.371086</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.022462</td>\n",
       "      <td>0.299013</td>\n",
       "      <td>8.0</td>\n",
       "      <td>-0.106883</td>\n",
       "      <td>0.283780</td>\n",
       "      <td>8.0</td>\n",
       "      <td>-0.112486</td>\n",
       "      <td>...</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.096579</td>\n",
       "      <td>0.337922</td>\n",
       "      <td>8.0</td>\n",
       "      <td>-0.622516</td>\n",
       "      <td>1.193240</td>\n",
       "      <td>8.0</td>\n",
       "      <td>-1.131078</td>\n",
       "      <td>1.266600</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.063138</td>\n",
       "      <td>0.389183</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.306976</td>\n",
       "      <td>0.461439</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.101113</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.099271</td>\n",
       "      <td>0.643307</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.146682</td>\n",
       "      <td>2.500634</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.983295</td>\n",
       "      <td>2.484216</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.134536</td>\n",
       "      <td>0.163912</td>\n",
       "      <td>10.0</td>\n",
       "      <td>-0.191888</td>\n",
       "      <td>0.164103</td>\n",
       "      <td>10.0</td>\n",
       "      <td>-0.368959</td>\n",
       "      <td>...</td>\n",
       "      <td>10.0</td>\n",
       "      <td>-0.307327</td>\n",
       "      <td>0.230147</td>\n",
       "      <td>10.0</td>\n",
       "      <td>-1.308992</td>\n",
       "      <td>0.439651</td>\n",
       "      <td>10.0</td>\n",
       "      <td>-1.288034</td>\n",
       "      <td>0.402243</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAED1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.569984</td>\n",
       "      <td>0.068138</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.581913</td>\n",
       "      <td>0.299041</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.365310</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.620177</td>\n",
       "      <td>0.667873</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-1.387034</td>\n",
       "      <td>0.493093</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-1.428928</td>\n",
       "      <td>0.662367</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAGAB</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.353526</td>\n",
       "      <td>0.204848</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.329347</td>\n",
       "      <td>0.488865</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.895738</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.019506</td>\n",
       "      <td>0.076190</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-1.145532</td>\n",
       "      <td>0.135048</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-1.032714</td>\n",
       "      <td>0.038582</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AARD</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.405413</td>\n",
       "      <td>0.009258</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.221888</td>\n",
       "      <td>0.012100</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.136083</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.273315</td>\n",
       "      <td>0.239510</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.322831</td>\n",
       "      <td>0.813271</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.064267</td>\n",
       "      <td>0.477512</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABHD16A</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.046190</td>\n",
       "      <td>0.309573</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.119689</td>\n",
       "      <td>0.428147</td>\n",
       "      <td>7.0</td>\n",
       "      <td>-0.168839</td>\n",
       "      <td>...</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.120121</td>\n",
       "      <td>0.366649</td>\n",
       "      <td>7.0</td>\n",
       "      <td>-0.542167</td>\n",
       "      <td>0.781300</td>\n",
       "      <td>7.0</td>\n",
       "      <td>-0.427547</td>\n",
       "      <td>0.752641</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABHD16B</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.002449</td>\n",
       "      <td>0.336360</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.262214</td>\n",
       "      <td>0.098413</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.142133</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.047215</td>\n",
       "      <td>0.136905</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.739588</td>\n",
       "      <td>0.099636</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.733932</td>\n",
       "      <td>0.162333</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABRACL</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.184052</td>\n",
       "      <td>0.369583</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.045057</td>\n",
       "      <td>0.449063</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.016780</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.180320</td>\n",
       "      <td>0.453096</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.329651</td>\n",
       "      <td>0.572447</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.338731</td>\n",
       "      <td>0.725840</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZC2HC1C</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.137910</td>\n",
       "      <td>0.173446</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.007626</td>\n",
       "      <td>0.249158</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.028270</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.049071</td>\n",
       "      <td>0.317767</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.743665</td>\n",
       "      <td>0.085947</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.613687</td>\n",
       "      <td>0.173659</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZFAND4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.093117</td>\n",
       "      <td>0.311586</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.271978</td>\n",
       "      <td>0.401376</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.380334</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.429613</td>\n",
       "      <td>0.615392</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.044634</td>\n",
       "      <td>0.537207</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.717845</td>\n",
       "      <td>0.644830</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZFHX2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.264805</td>\n",
       "      <td>0.153346</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.340443</td>\n",
       "      <td>0.175126</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.355297</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.110631</td>\n",
       "      <td>0.203873</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.634677</td>\n",
       "      <td>0.145443</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.704455</td>\n",
       "      <td>0.419995</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZFP62</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.313350</td>\n",
       "      <td>0.164382</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.272276</td>\n",
       "      <td>0.104812</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.234086</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.136079</td>\n",
       "      <td>0.083914</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.600790</td>\n",
       "      <td>0.228457</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.403734</td>\n",
       "      <td>0.239438</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZG16</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.076855</td>\n",
       "      <td>0.353263</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.044489</td>\n",
       "      <td>0.294222</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.169905</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.173585</td>\n",
       "      <td>0.320943</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.173767</td>\n",
       "      <td>0.284184</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.313039</td>\n",
       "      <td>0.315169</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZG16B</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.104232</td>\n",
       "      <td>0.295782</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.022509</td>\n",
       "      <td>0.161646</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.075338</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.107987</td>\n",
       "      <td>0.248984</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.930855</td>\n",
       "      <td>0.707679</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.779266</td>\n",
       "      <td>0.785660</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZGLP1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.013438</td>\n",
       "      <td>0.363166</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.038952</td>\n",
       "      <td>0.242658</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.232664</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.024691</td>\n",
       "      <td>0.316467</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.277580</td>\n",
       "      <td>0.575634</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.060090</td>\n",
       "      <td>0.463346</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZMYM6NB</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.053447</td>\n",
       "      <td>0.356120</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.038780</td>\n",
       "      <td>0.310126</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.242967</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.085448</td>\n",
       "      <td>0.206854</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.865397</td>\n",
       "      <td>0.253819</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.066366</td>\n",
       "      <td>0.226692</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF285</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.057408</td>\n",
       "      <td>0.405140</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.083629</td>\n",
       "      <td>0.426387</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.180654</td>\n",
       "      <td>...</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.111348</td>\n",
       "      <td>0.468329</td>\n",
       "      <td>7.0</td>\n",
       "      <td>-1.316528</td>\n",
       "      <td>0.890498</td>\n",
       "      <td>7.0</td>\n",
       "      <td>-1.178316</td>\n",
       "      <td>1.048426</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF296</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.074129</td>\n",
       "      <td>0.345844</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.099531</td>\n",
       "      <td>0.330443</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.369446</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.034401</td>\n",
       "      <td>0.399165</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.834741</td>\n",
       "      <td>0.723198</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.085035</td>\n",
       "      <td>0.796362</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF321P</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.028373</td>\n",
       "      <td>0.324011</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.015178</td>\n",
       "      <td>0.182710</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.135974</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.002291</td>\n",
       "      <td>0.491976</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.327449</td>\n",
       "      <td>1.246484</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.336306</td>\n",
       "      <td>1.445153</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF322</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.050622</td>\n",
       "      <td>0.994060</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.229144</td>\n",
       "      <td>0.680171</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.349549</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.128859</td>\n",
       "      <td>0.987379</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.649189</td>\n",
       "      <td>1.323274</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.153885</td>\n",
       "      <td>0.844021</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF526</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.011759</td>\n",
       "      <td>0.405006</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.021178</td>\n",
       "      <td>0.365149</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.327258</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.202848</td>\n",
       "      <td>0.579164</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.576374</td>\n",
       "      <td>0.277832</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.454169</td>\n",
       "      <td>0.386610</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF571</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.091387</td>\n",
       "      <td>0.178220</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.073208</td>\n",
       "      <td>0.442532</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.095097</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.168372</td>\n",
       "      <td>0.359044</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.446160</td>\n",
       "      <td>0.529834</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.494917</td>\n",
       "      <td>0.424566</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF593</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.183626</td>\n",
       "      <td>0.316487</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.301477</td>\n",
       "      <td>0.253294</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.341763</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.020286</td>\n",
       "      <td>0.313360</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.748938</td>\n",
       "      <td>0.291800</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.599245</td>\n",
       "      <td>0.337400</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF600</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.119798</td>\n",
       "      <td>0.472487</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.276862</td>\n",
       "      <td>0.357779</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.260818</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.219516</td>\n",
       "      <td>0.542814</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.560330</td>\n",
       "      <td>0.741100</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.474766</td>\n",
       "      <td>0.979289</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF658</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.553425</td>\n",
       "      <td>0.332304</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.384311</td>\n",
       "      <td>0.240181</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.620026</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.357992</td>\n",
       "      <td>0.574512</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.906369</td>\n",
       "      <td>0.723763</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.947172</td>\n",
       "      <td>0.407501</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF750</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.011203</td>\n",
       "      <td>0.105498</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.027877</td>\n",
       "      <td>0.139770</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.114199</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.086703</td>\n",
       "      <td>0.118125</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-1.000571</td>\n",
       "      <td>0.611606</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.958318</td>\n",
       "      <td>1.101352</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF784</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.065353</td>\n",
       "      <td>0.118628</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.024915</td>\n",
       "      <td>0.192403</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.016049</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.148791</td>\n",
       "      <td>0.082918</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.622454</td>\n",
       "      <td>0.139343</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.629896</td>\n",
       "      <td>0.141685</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF812</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.467668</td>\n",
       "      <td>0.007286</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.399851</td>\n",
       "      <td>0.141621</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.504722</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.307488</td>\n",
       "      <td>0.463714</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.084199</td>\n",
       "      <td>0.297511</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.208534</td>\n",
       "      <td>0.110100</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF816</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.241599</td>\n",
       "      <td>0.083833</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.454061</td>\n",
       "      <td>0.246946</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.185980</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.204872</td>\n",
       "      <td>0.259122</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.496710</td>\n",
       "      <td>0.115607</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.971898</td>\n",
       "      <td>0.659026</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF821</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.233381</td>\n",
       "      <td>0.142922</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-0.089025</td>\n",
       "      <td>0.128496</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-0.140905</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-0.159157</td>\n",
       "      <td>0.256497</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-0.978209</td>\n",
       "      <td>0.299940</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-1.381907</td>\n",
       "      <td>0.467752</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF878</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.341409</td>\n",
       "      <td>0.337638</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.376032</td>\n",
       "      <td>0.402329</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.321417</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.179532</td>\n",
       "      <td>0.109814</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.206510</td>\n",
       "      <td>0.336922</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.326782</td>\n",
       "      <td>0.225608</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF879</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.212938</td>\n",
       "      <td>0.109723</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.175749</td>\n",
       "      <td>0.127330</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.253557</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.069776</td>\n",
       "      <td>0.216324</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.126029</td>\n",
       "      <td>0.353817</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.337558</td>\n",
       "      <td>0.116496</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF880</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.141626</td>\n",
       "      <td>0.243410</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.192052</td>\n",
       "      <td>0.205433</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.327743</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.469206</td>\n",
       "      <td>0.295593</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-1.740418</td>\n",
       "      <td>0.299818</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-1.293956</td>\n",
       "      <td>0.025867</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNHIT2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.327564</td>\n",
       "      <td>0.186952</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.047789</td>\n",
       "      <td>0.133599</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.476902</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.038725</td>\n",
       "      <td>0.212157</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-1.480483</td>\n",
       "      <td>0.354998</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-1.575927</td>\n",
       "      <td>0.104713</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZP4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.329607</td>\n",
       "      <td>0.454184</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.349568</td>\n",
       "      <td>0.210354</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.435462</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.220061</td>\n",
       "      <td>0.252855</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.068949</td>\n",
       "      <td>0.127473</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.038344</td>\n",
       "      <td>0.432609</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN30</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.070923</td>\n",
       "      <td>0.263148</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.308303</td>\n",
       "      <td>0.184020</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.217292</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.278542</td>\n",
       "      <td>0.203566</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.563180</td>\n",
       "      <td>0.138333</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.603949</td>\n",
       "      <td>0.316927</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDA</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.516810</td>\n",
       "      <td>0.048150</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.451882</td>\n",
       "      <td>0.151484</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.527109</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.204421</td>\n",
       "      <td>0.175504</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.223882</td>\n",
       "      <td>0.537907</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.347043</td>\n",
       "      <td>0.217007</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hsa-mir-3187</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.283218</td>\n",
       "      <td>0.430932</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.162773</td>\n",
       "      <td>0.451196</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.733347</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.252364</td>\n",
       "      <td>0.402374</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.753625</td>\n",
       "      <td>0.273931</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-1.477145</td>\n",
       "      <td>0.096784</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2151 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                CNV         norm_count_D7_Rep1                  \\\n",
       "             median std len             median       std   len   \n",
       "1-Dec           NaN NaN NaN          -0.096822  0.309539   3.0   \n",
       "1-Mar           NaN NaN NaN          -0.001451  0.149900   9.0   \n",
       "1-Sep           NaN NaN NaN          -0.250745  0.104664   3.0   \n",
       "10-Mar          NaN NaN NaN           0.316717  0.359520   4.0   \n",
       "10-Sep          NaN NaN NaN          -0.541840  0.514727   3.0   \n",
       "11-Mar          NaN NaN NaN           0.120893  0.221079   3.0   \n",
       "11-Sep          NaN NaN NaN           0.133383  0.116819   3.0   \n",
       "12-Sep          NaN NaN NaN           0.005727  0.072789   3.0   \n",
       "14-Sep          NaN NaN NaN          -0.290554  0.306691   3.0   \n",
       "2-Mar           NaN NaN NaN           0.242636  0.223773  11.0   \n",
       "2-Sep           NaN NaN NaN          -0.074227  0.149550   4.0   \n",
       "3-Mar           NaN NaN NaN          -0.077972  0.152052   4.0   \n",
       "3-Sep           NaN NaN NaN           0.046410  0.156341   4.0   \n",
       "4-Mar           NaN NaN NaN           0.200761  0.241437   5.0   \n",
       "4-Sep           NaN NaN NaN           0.073473  0.370888   8.0   \n",
       "5-Mar           NaN NaN NaN           0.364824  0.234827   3.0   \n",
       "5-Sep           NaN NaN NaN          -0.204853  0.070917   5.0   \n",
       "6-Mar           NaN NaN NaN          -0.080671  0.646677   4.0   \n",
       "6-Sep           NaN NaN NaN           0.316024  0.114487   4.0   \n",
       "7-Mar           NaN NaN NaN          -0.103194  0.370343   6.0   \n",
       "8-Mar           NaN NaN NaN          -0.043177  0.139767   4.0   \n",
       "8-Sep           NaN NaN NaN          -0.022462  0.299013   8.0   \n",
       "9-Mar           NaN NaN NaN          -0.063138  0.389183   4.0   \n",
       "9-Sep           NaN NaN NaN          -0.134536  0.163912  10.0   \n",
       "AAED1           NaN NaN NaN          -0.569984  0.068138   2.0   \n",
       "AAGAB           NaN NaN NaN          -0.353526  0.204848   2.0   \n",
       "AARD            NaN NaN NaN           0.405413  0.009258   3.0   \n",
       "ABHD16A         NaN NaN NaN           0.046190  0.309573   7.0   \n",
       "ABHD16B         NaN NaN NaN          -0.002449  0.336360   2.0   \n",
       "ABRACL          NaN NaN NaN          -0.184052  0.369583   4.0   \n",
       "...             ...  ..  ..                ...       ...   ...   \n",
       "ZC2HC1C         NaN NaN NaN           0.137910  0.173446   4.0   \n",
       "ZFAND4          NaN NaN NaN          -0.093117  0.311586   4.0   \n",
       "ZFHX2           NaN NaN NaN           0.264805  0.153346   4.0   \n",
       "ZFP62           NaN NaN NaN           0.313350  0.164382   2.0   \n",
       "ZG16            NaN NaN NaN          -0.076855  0.353263   4.0   \n",
       "ZG16B           NaN NaN NaN           0.104232  0.295782   4.0   \n",
       "ZGLP1           NaN NaN NaN          -0.013438  0.363166   4.0   \n",
       "ZMYM6NB         NaN NaN NaN          -0.053447  0.356120   4.0   \n",
       "ZNF285          NaN NaN NaN           0.057408  0.405140   7.0   \n",
       "ZNF296          NaN NaN NaN           0.074129  0.345844   3.0   \n",
       "ZNF321P         NaN NaN NaN           0.028373  0.324011   3.0   \n",
       "ZNF322          NaN NaN NaN          -0.050622  0.994060   2.0   \n",
       "ZNF526          NaN NaN NaN          -0.011759  0.405006   2.0   \n",
       "ZNF571          NaN NaN NaN           0.091387  0.178220   2.0   \n",
       "ZNF593          NaN NaN NaN          -0.183626  0.316487   4.0   \n",
       "ZNF600          NaN NaN NaN           0.119798  0.472487   2.0   \n",
       "ZNF658          NaN NaN NaN          -0.553425  0.332304   3.0   \n",
       "ZNF750          NaN NaN NaN          -0.011203  0.105498   4.0   \n",
       "ZNF784          NaN NaN NaN           0.065353  0.118628   3.0   \n",
       "ZNF812          NaN NaN NaN           0.467668  0.007286   2.0   \n",
       "ZNF816          NaN NaN NaN           0.241599  0.083833   2.0   \n",
       "ZNF821          NaN NaN NaN          -0.233381  0.142922   5.0   \n",
       "ZNF878          NaN NaN NaN           0.341409  0.337638   2.0   \n",
       "ZNF879          NaN NaN NaN           0.212938  0.109723   3.0   \n",
       "ZNF880          NaN NaN NaN          -0.141626  0.243410   2.0   \n",
       "ZNHIT2          NaN NaN NaN          -0.327564  0.186952   2.0   \n",
       "ZP4             NaN NaN NaN          -0.329607  0.454184   3.0   \n",
       "ZSCAN30         NaN NaN NaN           0.070923  0.263148   3.0   \n",
       "ZXDA            NaN NaN NaN           0.516810  0.048150   2.0   \n",
       "hsa-mir-3187    NaN NaN NaN          -0.283218  0.430932   3.0   \n",
       "\n",
       "             norm_count_D7_Rep2                 norm_count_D14_Rep1  ...   \\\n",
       "                         median       std   len              median  ...    \n",
       "1-Dec                 -0.313602  0.252063   3.0           -0.156710  ...    \n",
       "1-Mar                 -0.007358  0.238821   9.0            0.144568  ...    \n",
       "1-Sep                 -0.154311  0.324163   3.0           -0.389285  ...    \n",
       "10-Mar                 0.290522  0.378278   4.0            0.143517  ...    \n",
       "10-Sep                -0.460078  0.578075   3.0           -0.547001  ...    \n",
       "11-Mar                 0.230257  0.336709   3.0            0.159451  ...    \n",
       "11-Sep                 0.077200  0.065924   3.0            0.331909  ...    \n",
       "12-Sep                -0.032687  0.133818   3.0            0.037297  ...    \n",
       "14-Sep                -0.142270  0.413997   3.0           -0.089395  ...    \n",
       "2-Mar                  0.052407  0.250299  11.0            0.300373  ...    \n",
       "2-Sep                 -0.067209  0.195979   4.0            0.077875  ...    \n",
       "3-Mar                 -0.130657  0.307455   4.0           -0.022833  ...    \n",
       "3-Sep                  0.008428  0.260490   4.0            0.093611  ...    \n",
       "4-Mar                  0.188366  0.138867   5.0           -0.220544  ...    \n",
       "4-Sep                  0.071134  0.171446   8.0            0.021822  ...    \n",
       "5-Mar                  0.270827  0.102993   3.0            0.180895  ...    \n",
       "5-Sep                 -0.166603  0.166208   5.0           -0.363972  ...    \n",
       "6-Mar                  0.081839  0.491706   4.0            0.095961  ...    \n",
       "6-Sep                  0.208226  0.163103   4.0            0.503353  ...    \n",
       "7-Mar                 -0.141245  0.247514   6.0            0.176513  ...    \n",
       "8-Mar                 -0.022048  0.033383   4.0           -0.044235  ...    \n",
       "8-Sep                 -0.106883  0.283780   8.0           -0.112486  ...    \n",
       "9-Mar                 -0.306976  0.461439   4.0           -0.101113  ...    \n",
       "9-Sep                 -0.191888  0.164103  10.0           -0.368959  ...    \n",
       "AAED1                 -0.581913  0.299041   2.0           -0.365310  ...    \n",
       "AAGAB                 -0.329347  0.488865   2.0           -0.895738  ...    \n",
       "AARD                   0.221888  0.012100   3.0            0.136083  ...    \n",
       "ABHD16A                0.119689  0.428147   7.0           -0.168839  ...    \n",
       "ABHD16B                0.262214  0.098413   2.0            0.142133  ...    \n",
       "ABRACL                 0.045057  0.449063   4.0            0.016780  ...    \n",
       "...                         ...       ...   ...                 ...  ...    \n",
       "ZC2HC1C                0.007626  0.249158   4.0           -0.028270  ...    \n",
       "ZFAND4                -0.271978  0.401376   4.0           -0.380334  ...    \n",
       "ZFHX2                  0.340443  0.175126   4.0            0.355297  ...    \n",
       "ZFP62                  0.272276  0.104812   2.0            0.234086  ...    \n",
       "ZG16                   0.044489  0.294222   4.0           -0.169905  ...    \n",
       "ZG16B                  0.022509  0.161646   4.0            0.075338  ...    \n",
       "ZGLP1                  0.038952  0.242658   4.0           -0.232664  ...    \n",
       "ZMYM6NB               -0.038780  0.310126   4.0           -0.242967  ...    \n",
       "ZNF285                 0.083629  0.426387   7.0            0.180654  ...    \n",
       "ZNF296                 0.099531  0.330443   3.0            0.369446  ...    \n",
       "ZNF321P               -0.015178  0.182710   3.0            0.135974  ...    \n",
       "ZNF322                -0.229144  0.680171   2.0           -0.349549  ...    \n",
       "ZNF526                 0.021178  0.365149   2.0           -0.327258  ...    \n",
       "ZNF571                 0.073208  0.442532   2.0            0.095097  ...    \n",
       "ZNF593                -0.301477  0.253294   4.0           -0.341763  ...    \n",
       "ZNF600                 0.276862  0.357779   2.0           -0.260818  ...    \n",
       "ZNF658                -0.384311  0.240181   3.0           -0.620026  ...    \n",
       "ZNF750                 0.027877  0.139770   4.0            0.114199  ...    \n",
       "ZNF784                 0.024915  0.192403   3.0           -0.016049  ...    \n",
       "ZNF812                 0.399851  0.141621   2.0            0.504722  ...    \n",
       "ZNF816                 0.454061  0.246946   2.0            0.185980  ...    \n",
       "ZNF821                -0.089025  0.128496   5.0           -0.140905  ...    \n",
       "ZNF878                 0.376032  0.402329   2.0            0.321417  ...    \n",
       "ZNF879                 0.175749  0.127330   3.0            0.253557  ...    \n",
       "ZNF880                -0.192052  0.205433   2.0           -0.327743  ...    \n",
       "ZNHIT2                -0.047789  0.133599   2.0           -0.476902  ...    \n",
       "ZP4                   -0.349568  0.210354   3.0           -0.435462  ...    \n",
       "ZSCAN30                0.308303  0.184020   3.0            0.217292  ...    \n",
       "ZXDA                   0.451882  0.151484   2.0            0.527109  ...    \n",
       "hsa-mir-3187          -0.162773  0.451196   3.0           -0.733347  ...    \n",
       "\n",
       "             norm_count_PLX7_Rep1 norm_count_PLX7_Rep2                  \\\n",
       "                              len               median       std   len   \n",
       "1-Dec                         3.0            -0.142697  0.429846   3.0   \n",
       "1-Mar                         9.0            -0.160655  0.428572   9.0   \n",
       "1-Sep                         3.0            -0.164021  0.452095   3.0   \n",
       "10-Mar                        4.0            -0.014089  0.283933   4.0   \n",
       "10-Sep                        3.0            -0.197123  0.467431   3.0   \n",
       "11-Mar                        3.0             0.101848  0.155371   3.0   \n",
       "11-Sep                        3.0            -0.076066  0.004554   3.0   \n",
       "12-Sep                        3.0             0.039585  0.109875   3.0   \n",
       "14-Sep                        3.0            -0.594866  0.327882   3.0   \n",
       "2-Mar                        11.0             0.115797  0.335042  11.0   \n",
       "2-Sep                         4.0            -0.071584  0.121537   4.0   \n",
       "3-Mar                         4.0            -0.116003  0.218735   4.0   \n",
       "3-Sep                         4.0             0.008398  0.267759   4.0   \n",
       "4-Mar                         5.0             0.150098  0.290600   5.0   \n",
       "4-Sep                         8.0             0.019595  0.317571   8.0   \n",
       "5-Mar                         3.0             0.107403  0.455855   3.0   \n",
       "5-Sep                         5.0            -0.139331  0.124380   5.0   \n",
       "6-Mar                         4.0            -0.334222  0.419352   4.0   \n",
       "6-Sep                         4.0             0.252912  0.116103   4.0   \n",
       "7-Mar                         6.0            -0.219990  0.269957   6.0   \n",
       "8-Mar                         4.0             0.002691  0.121394   4.0   \n",
       "8-Sep                         8.0             0.096579  0.337922   8.0   \n",
       "9-Mar                         4.0            -0.099271  0.643307   4.0   \n",
       "9-Sep                        10.0            -0.307327  0.230147  10.0   \n",
       "AAED1                         2.0            -0.620177  0.667873   2.0   \n",
       "AAGAB                         2.0            -0.019506  0.076190   2.0   \n",
       "AARD                          3.0             0.273315  0.239510   3.0   \n",
       "ABHD16A                       7.0             0.120121  0.366649   7.0   \n",
       "ABHD16B                       2.0            -0.047215  0.136905   2.0   \n",
       "ABRACL                        4.0            -0.180320  0.453096   4.0   \n",
       "...                           ...                  ...       ...   ...   \n",
       "ZC2HC1C                       4.0             0.049071  0.317767   4.0   \n",
       "ZFAND4                        4.0            -0.429613  0.615392   4.0   \n",
       "ZFHX2                         4.0             0.110631  0.203873   4.0   \n",
       "ZFP62                         2.0             0.136079  0.083914   2.0   \n",
       "ZG16                          4.0            -0.173585  0.320943   4.0   \n",
       "ZG16B                         4.0            -0.107987  0.248984   4.0   \n",
       "ZGLP1                         4.0            -0.024691  0.316467   4.0   \n",
       "ZMYM6NB                       4.0            -0.085448  0.206854   4.0   \n",
       "ZNF285                        7.0             0.111348  0.468329   7.0   \n",
       "ZNF296                        3.0             0.034401  0.399165   3.0   \n",
       "ZNF321P                       3.0             0.002291  0.491976   3.0   \n",
       "ZNF322                        2.0            -0.128859  0.987379   2.0   \n",
       "ZNF526                        2.0             0.202848  0.579164   2.0   \n",
       "ZNF571                        2.0             0.168372  0.359044   2.0   \n",
       "ZNF593                        4.0             0.020286  0.313360   4.0   \n",
       "ZNF600                        2.0             0.219516  0.542814   2.0   \n",
       "ZNF658                        3.0            -0.357992  0.574512   3.0   \n",
       "ZNF750                        4.0            -0.086703  0.118125   4.0   \n",
       "ZNF784                        3.0             0.148791  0.082918   3.0   \n",
       "ZNF812                        2.0             0.307488  0.463714   2.0   \n",
       "ZNF816                        2.0             0.204872  0.259122   2.0   \n",
       "ZNF821                        5.0            -0.159157  0.256497   5.0   \n",
       "ZNF878                        2.0             0.179532  0.109814   2.0   \n",
       "ZNF879                        3.0             0.069776  0.216324   3.0   \n",
       "ZNF880                        2.0            -0.469206  0.295593   2.0   \n",
       "ZNHIT2                        2.0            -0.038725  0.212157   2.0   \n",
       "ZP4                           3.0            -0.220061  0.252855   3.0   \n",
       "ZSCAN30                       3.0             0.278542  0.203566   3.0   \n",
       "ZXDA                          2.0             0.204421  0.175504   2.0   \n",
       "hsa-mir-3187                  3.0            -0.252364  0.402374   3.0   \n",
       "\n",
       "             norm_count_PLX14_Rep1                 norm_count_PLX14_Rep2  \\\n",
       "                            median       std   len                median   \n",
       "1-Dec                    -0.738297  0.488660   3.0             -0.649533   \n",
       "1-Mar                    -1.105906  0.597998   9.0             -1.053001   \n",
       "1-Sep                    -1.263459  0.374423   3.0             -1.215216   \n",
       "10-Mar                   -1.155326  0.484769   4.0             -0.987062   \n",
       "10-Sep                   -0.800630  0.352074   3.0             -1.109213   \n",
       "11-Mar                   -0.509583  0.633365   3.0             -0.215090   \n",
       "11-Sep                   -0.932400  0.274641   3.0             -1.244964   \n",
       "12-Sep                   -1.450022  0.122485   3.0             -1.377543   \n",
       "14-Sep                   -1.547629  0.321109   3.0             -1.709218   \n",
       "2-Mar                    -0.788051  0.716257  11.0             -0.852748   \n",
       "2-Sep                    -1.473724  0.705837   4.0             -1.435851   \n",
       "3-Mar                    -0.742879  0.584929   4.0             -0.685291   \n",
       "3-Sep                    -1.007701  0.547351   4.0             -1.218126   \n",
       "4-Mar                    -0.932872  0.731989   5.0             -1.350516   \n",
       "4-Sep                    -0.877557  0.502653   8.0             -0.840931   \n",
       "5-Mar                    -1.999522  1.551640   3.0             -1.627167   \n",
       "5-Sep                    -0.927778  0.431610   5.0             -1.396363   \n",
       "6-Mar                    -1.136917  0.241243   4.0             -1.248705   \n",
       "6-Sep                    -0.490197  0.334553   4.0             -0.409739   \n",
       "7-Mar                    -0.953055  0.356929   6.0             -0.962991   \n",
       "8-Mar                    -1.016384  0.570748   4.0             -1.066330   \n",
       "8-Sep                    -0.622516  1.193240   8.0             -1.131078   \n",
       "9-Mar                    -1.146682  2.500634   4.0             -0.983295   \n",
       "9-Sep                    -1.308992  0.439651  10.0             -1.288034   \n",
       "AAED1                    -1.387034  0.493093   2.0             -1.428928   \n",
       "AAGAB                    -1.145532  0.135048   2.0             -1.032714   \n",
       "AARD                     -0.322831  0.813271   3.0              0.064267   \n",
       "ABHD16A                  -0.542167  0.781300   7.0             -0.427547   \n",
       "ABHD16B                  -0.739588  0.099636   2.0             -0.733932   \n",
       "ABRACL                   -1.329651  0.572447   4.0             -1.338731   \n",
       "...                            ...       ...   ...                   ...   \n",
       "ZC2HC1C                  -0.743665  0.085947   4.0             -0.613687   \n",
       "ZFAND4                   -1.044634  0.537207   4.0             -0.717845   \n",
       "ZFHX2                    -0.634677  0.145443   4.0             -0.704455   \n",
       "ZFP62                    -0.600790  0.228457   2.0             -0.403734   \n",
       "ZG16                     -1.173767  0.284184   4.0             -1.313039   \n",
       "ZG16B                    -0.930855  0.707679   4.0             -0.779266   \n",
       "ZGLP1                    -1.277580  0.575634   4.0             -1.060090   \n",
       "ZMYM6NB                  -0.865397  0.253819   4.0             -1.066366   \n",
       "ZNF285                   -1.316528  0.890498   7.0             -1.178316   \n",
       "ZNF296                   -0.834741  0.723198   3.0             -1.085035   \n",
       "ZNF321P                  -0.327449  1.246484   3.0             -0.336306   \n",
       "ZNF322                   -0.649189  1.323274   2.0             -0.153885   \n",
       "ZNF526                   -0.576374  0.277832   2.0             -0.454169   \n",
       "ZNF571                    0.446160  0.529834   2.0              0.494917   \n",
       "ZNF593                   -0.748938  0.291800   4.0             -0.599245   \n",
       "ZNF600                   -0.560330  0.741100   2.0             -0.474766   \n",
       "ZNF658                   -0.906369  0.723763   3.0             -0.947172   \n",
       "ZNF750                   -1.000571  0.611606   4.0             -0.958318   \n",
       "ZNF784                   -0.622454  0.139343   3.0             -0.629896   \n",
       "ZNF812                    0.084199  0.297511   2.0             -0.208534   \n",
       "ZNF816                   -0.496710  0.115607   2.0             -0.971898   \n",
       "ZNF821                   -0.978209  0.299940   5.0             -1.381907   \n",
       "ZNF878                   -0.206510  0.336922   2.0             -0.326782   \n",
       "ZNF879                   -0.126029  0.353817   3.0             -0.337558   \n",
       "ZNF880                   -1.740418  0.299818   2.0             -1.293956   \n",
       "ZNHIT2                   -1.480483  0.354998   2.0             -1.575927   \n",
       "ZP4                      -1.068949  0.127473   3.0             -1.038344   \n",
       "ZSCAN30                  -0.563180  0.138333   3.0             -0.603949   \n",
       "ZXDA                     -0.223882  0.537907   2.0             -0.347043   \n",
       "hsa-mir-3187             -1.753625  0.273931   3.0             -1.477145   \n",
       "\n",
       "                              \n",
       "                   std   len  \n",
       "1-Dec         0.582200   3.0  \n",
       "1-Mar         0.656827   9.0  \n",
       "1-Sep         0.361984   3.0  \n",
       "10-Mar        0.601717   4.0  \n",
       "10-Sep        0.546510   3.0  \n",
       "11-Mar        0.983348   3.0  \n",
       "11-Sep        0.272671   3.0  \n",
       "12-Sep        0.146387   3.0  \n",
       "14-Sep        0.409802   3.0  \n",
       "2-Mar         0.710373  11.0  \n",
       "2-Sep         0.480136   4.0  \n",
       "3-Mar         0.749166   4.0  \n",
       "3-Sep         0.494752   4.0  \n",
       "4-Mar         0.870646   5.0  \n",
       "4-Sep         0.603345   8.0  \n",
       "5-Mar         1.267703   3.0  \n",
       "5-Sep         0.408007   5.0  \n",
       "6-Mar         0.274294   4.0  \n",
       "6-Sep         0.386353   4.0  \n",
       "7-Mar         0.292234   6.0  \n",
       "8-Mar         0.371086   4.0  \n",
       "8-Sep         1.266600   8.0  \n",
       "9-Mar         2.484216   4.0  \n",
       "9-Sep         0.402243  10.0  \n",
       "AAED1         0.662367   2.0  \n",
       "AAGAB         0.038582   2.0  \n",
       "AARD          0.477512   3.0  \n",
       "ABHD16A       0.752641   7.0  \n",
       "ABHD16B       0.162333   2.0  \n",
       "ABRACL        0.725840   4.0  \n",
       "...                ...   ...  \n",
       "ZC2HC1C       0.173659   4.0  \n",
       "ZFAND4        0.644830   4.0  \n",
       "ZFHX2         0.419995   4.0  \n",
       "ZFP62         0.239438   2.0  \n",
       "ZG16          0.315169   4.0  \n",
       "ZG16B         0.785660   4.0  \n",
       "ZGLP1         0.463346   4.0  \n",
       "ZMYM6NB       0.226692   4.0  \n",
       "ZNF285        1.048426   7.0  \n",
       "ZNF296        0.796362   3.0  \n",
       "ZNF321P       1.445153   3.0  \n",
       "ZNF322        0.844021   2.0  \n",
       "ZNF526        0.386610   2.0  \n",
       "ZNF571        0.424566   2.0  \n",
       "ZNF593        0.337400   4.0  \n",
       "ZNF600        0.979289   2.0  \n",
       "ZNF658        0.407501   3.0  \n",
       "ZNF750        1.101352   4.0  \n",
       "ZNF784        0.141685   3.0  \n",
       "ZNF812        0.110100   2.0  \n",
       "ZNF816        0.659026   2.0  \n",
       "ZNF821        0.467752   5.0  \n",
       "ZNF878        0.225608   2.0  \n",
       "ZNF879        0.116496   3.0  \n",
       "ZNF880        0.025867   2.0  \n",
       "ZNHIT2        0.104713   2.0  \n",
       "ZP4           0.432609   3.0  \n",
       "ZSCAN30       0.316927   3.0  \n",
       "ZXDA          0.217007   2.0  \n",
       "hsa-mir-3187  0.096784   3.0  \n",
       "\n",
       "[2151 rows x 27 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Which genes are not in the CNV data?\n",
    "cnv_and_guides[numpy.isnan(cnv_and_guides[\"CNV\"][\"median\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"3\" halign=\"left\">CNV</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_D7_Rep1</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_D7_Rep2</th>\n",
       "      <th>norm_count_D14_Rep1</th>\n",
       "      <th>...</th>\n",
       "      <th>norm_count_PLX7_Rep1</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_PLX7_Rep2</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_PLX14_Rep1</th>\n",
       "      <th colspan=\"3\" halign=\"left\">norm_count_PLX14_Rep2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>...</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>len</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2BP1</th>\n",
       "      <td>-0.005361</td>\n",
       "      <td>0.156898</td>\n",
       "      <td>343.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A3GALT2</th>\n",
       "      <td>0.082832</td>\n",
       "      <td>0.212417</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACSL</th>\n",
       "      <td>0.057494</td>\n",
       "      <td>0.171914</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AADACL1</th>\n",
       "      <td>-0.165770</td>\n",
       "      <td>0.140979</td>\n",
       "      <td>15.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABCA11P</th>\n",
       "      <td>0.030222</td>\n",
       "      <td>0.228350</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABCA17P</th>\n",
       "      <td>0.134179</td>\n",
       "      <td>0.123823</td>\n",
       "      <td>13.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABCC13</th>\n",
       "      <td>-0.207125</td>\n",
       "      <td>0.071016</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABCC6P1</th>\n",
       "      <td>-0.120501</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABO</th>\n",
       "      <td>0.216423</td>\n",
       "      <td>0.108170</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACCN1</th>\n",
       "      <td>-0.012577</td>\n",
       "      <td>0.152517</td>\n",
       "      <td>220.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACCN2</th>\n",
       "      <td>0.177619</td>\n",
       "      <td>0.270185</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACCN4</th>\n",
       "      <td>0.295845</td>\n",
       "      <td>0.188711</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACCN5</th>\n",
       "      <td>-0.201665</td>\n",
       "      <td>0.185809</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTL7A</th>\n",
       "      <td>0.420039</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTL7B</th>\n",
       "      <td>0.339726</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTN3</th>\n",
       "      <td>0.153029</td>\n",
       "      <td>0.125531</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTR2</th>\n",
       "      <td>0.164306</td>\n",
       "      <td>0.156949</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ADAM3A</th>\n",
       "      <td>-0.863125</td>\n",
       "      <td>0.439798</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ADAM5P</th>\n",
       "      <td>-0.195959</td>\n",
       "      <td>0.455284</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ADFP</th>\n",
       "      <td>0.062072</td>\n",
       "      <td>0.102746</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ADH1C</th>\n",
       "      <td>-0.341028</td>\n",
       "      <td>0.203640</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ADRA2C</th>\n",
       "      <td>0.319492</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFG3L1</th>\n",
       "      <td>0.120819</td>\n",
       "      <td>0.099450</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFM</th>\n",
       "      <td>-0.167160</td>\n",
       "      <td>0.350376</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AGAP11</th>\n",
       "      <td>0.082655</td>\n",
       "      <td>0.064363</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AHCTF1</th>\n",
       "      <td>0.264479</td>\n",
       "      <td>0.151891</td>\n",
       "      <td>18.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AHDC1</th>\n",
       "      <td>0.261548</td>\n",
       "      <td>0.134500</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AIM1L</th>\n",
       "      <td>0.140806</td>\n",
       "      <td>0.127140</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AK3L1</th>\n",
       "      <td>0.094108</td>\n",
       "      <td>0.120256</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKAP11</th>\n",
       "      <td>-0.096526</td>\n",
       "      <td>0.072024</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF676</th>\n",
       "      <td>-0.135092</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF678</th>\n",
       "      <td>0.237430</td>\n",
       "      <td>0.134827</td>\n",
       "      <td>14.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF701</th>\n",
       "      <td>0.047407</td>\n",
       "      <td>0.096168</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF702P</th>\n",
       "      <td>0.097253</td>\n",
       "      <td>0.192007</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF706</th>\n",
       "      <td>0.000613</td>\n",
       "      <td>0.118897</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF716</th>\n",
       "      <td>-0.096542</td>\n",
       "      <td>0.129688</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF717</th>\n",
       "      <td>-0.145257</td>\n",
       "      <td>0.054153</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF718</th>\n",
       "      <td>0.019428</td>\n",
       "      <td>0.179064</td>\n",
       "      <td>13.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF721</th>\n",
       "      <td>0.094312</td>\n",
       "      <td>0.188507</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF735</th>\n",
       "      <td>-0.008379</td>\n",
       "      <td>0.232152</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF738</th>\n",
       "      <td>0.021843</td>\n",
       "      <td>0.058719</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF761</th>\n",
       "      <td>-0.044199</td>\n",
       "      <td>0.074071</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF763</th>\n",
       "      <td>0.265358</td>\n",
       "      <td>0.056093</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF765</th>\n",
       "      <td>-0.076262</td>\n",
       "      <td>0.397507</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF767</th>\n",
       "      <td>-0.079668</td>\n",
       "      <td>0.200021</td>\n",
       "      <td>15.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF788</th>\n",
       "      <td>0.080578</td>\n",
       "      <td>0.217634</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF80</th>\n",
       "      <td>-0.041998</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF815</th>\n",
       "      <td>0.154976</td>\n",
       "      <td>0.159028</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF816A</th>\n",
       "      <td>-0.124984</td>\n",
       "      <td>0.102720</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF826</th>\n",
       "      <td>0.115634</td>\n",
       "      <td>0.104434</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF828</th>\n",
       "      <td>-0.038841</td>\n",
       "      <td>0.272088</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF830</th>\n",
       "      <td>0.000967</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF833</th>\n",
       "      <td>0.319374</td>\n",
       "      <td>0.092108</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF837</th>\n",
       "      <td>0.198976</td>\n",
       "      <td>0.415677</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF860</th>\n",
       "      <td>-0.148138</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF99</th>\n",
       "      <td>0.179160</td>\n",
       "      <td>0.232013</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN12</th>\n",
       "      <td>-0.035386</td>\n",
       "      <td>0.113249</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSCAN12L1</th>\n",
       "      <td>-0.081045</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>psiTPTE22</th>\n",
       "      <td>-0.085944</td>\n",
       "      <td>0.097844</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tAKR</th>\n",
       "      <td>-0.059410</td>\n",
       "      <td>0.082184</td>\n",
       "      <td>9.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2540 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                CNV                  norm_count_D7_Rep1          \\\n",
       "             median       std    len             median std len   \n",
       "A2BP1     -0.005361  0.156898  343.0                NaN NaN NaN   \n",
       "A3GALT2    0.082832  0.212417    2.0                NaN NaN NaN   \n",
       "AACSL      0.057494  0.171914    2.0                NaN NaN NaN   \n",
       "AADACL1   -0.165770  0.140979   15.0                NaN NaN NaN   \n",
       "ABCA11P    0.030222  0.228350    7.0                NaN NaN NaN   \n",
       "ABCA17P    0.134179  0.123823   13.0                NaN NaN NaN   \n",
       "ABCC13    -0.207125  0.071016    7.0                NaN NaN NaN   \n",
       "ABCC6P1   -0.120501       NaN    1.0                NaN NaN NaN   \n",
       "ABO        0.216423  0.108170    4.0                NaN NaN NaN   \n",
       "ACCN1     -0.012577  0.152517  220.0                NaN NaN NaN   \n",
       "ACCN2      0.177619  0.270185    5.0                NaN NaN NaN   \n",
       "ACCN4      0.295845  0.188711    6.0                NaN NaN NaN   \n",
       "ACCN5     -0.201665  0.185809   12.0                NaN NaN NaN   \n",
       "ACTL7A     0.420039       NaN    1.0                NaN NaN NaN   \n",
       "ACTL7B     0.339726       NaN    1.0                NaN NaN NaN   \n",
       "ACTN3      0.153029  0.125531    3.0                NaN NaN NaN   \n",
       "ACTR2      0.164306  0.156949    7.0                NaN NaN NaN   \n",
       "ADAM3A    -0.863125  0.439798   12.0                NaN NaN NaN   \n",
       "ADAM5P    -0.195959  0.455284   16.0                NaN NaN NaN   \n",
       "ADFP       0.062072  0.102746    2.0                NaN NaN NaN   \n",
       "ADH1C     -0.341028  0.203640    4.0                NaN NaN NaN   \n",
       "ADRA2C     0.319492       NaN    1.0                NaN NaN NaN   \n",
       "AFG3L1     0.120819  0.099450    7.0                NaN NaN NaN   \n",
       "AFM       -0.167160  0.350376    7.0                NaN NaN NaN   \n",
       "AGAP11     0.082655  0.064363    5.0                NaN NaN NaN   \n",
       "AHCTF1     0.264479  0.151891   18.0                NaN NaN NaN   \n",
       "AHDC1      0.261548  0.134500   16.0                NaN NaN NaN   \n",
       "AIM1L      0.140806  0.127140    4.0                NaN NaN NaN   \n",
       "AK3L1      0.094108  0.120256   16.0                NaN NaN NaN   \n",
       "AKAP11    -0.096526  0.072024   12.0                NaN NaN NaN   \n",
       "...             ...       ...    ...                ...  ..  ..   \n",
       "ZNF676    -0.135092       NaN    1.0                NaN NaN NaN   \n",
       "ZNF678     0.237430  0.134827   14.0                NaN NaN NaN   \n",
       "ZNF701     0.047407  0.096168    3.0                NaN NaN NaN   \n",
       "ZNF702P    0.097253  0.192007    4.0                NaN NaN NaN   \n",
       "ZNF706     0.000613  0.118897    4.0                NaN NaN NaN   \n",
       "ZNF716    -0.096542  0.129688    5.0                NaN NaN NaN   \n",
       "ZNF717    -0.145257  0.054153    2.0                NaN NaN NaN   \n",
       "ZNF718     0.019428  0.179064   13.0                NaN NaN NaN   \n",
       "ZNF721     0.094312  0.188507    4.0                NaN NaN NaN   \n",
       "ZNF735    -0.008379  0.232152    2.0                NaN NaN NaN   \n",
       "ZNF738     0.021843  0.058719    4.0                NaN NaN NaN   \n",
       "ZNF761    -0.044199  0.074071    4.0                NaN NaN NaN   \n",
       "ZNF763     0.265358  0.056093    2.0                NaN NaN NaN   \n",
       "ZNF765    -0.076262  0.397507    5.0                NaN NaN NaN   \n",
       "ZNF767    -0.079668  0.200021   15.0                NaN NaN NaN   \n",
       "ZNF788     0.080578  0.217634    3.0                NaN NaN NaN   \n",
       "ZNF80     -0.041998       NaN    1.0                NaN NaN NaN   \n",
       "ZNF815     0.154976  0.159028    4.0                NaN NaN NaN   \n",
       "ZNF816A   -0.124984  0.102720    2.0                NaN NaN NaN   \n",
       "ZNF826     0.115634  0.104434    6.0                NaN NaN NaN   \n",
       "ZNF828    -0.038841  0.272088    4.0                NaN NaN NaN   \n",
       "ZNF830     0.000967       NaN    1.0                NaN NaN NaN   \n",
       "ZNF833     0.319374  0.092108    4.0                NaN NaN NaN   \n",
       "ZNF837     0.198976  0.415677    3.0                NaN NaN NaN   \n",
       "ZNF860    -0.148138       NaN    1.0                NaN NaN NaN   \n",
       "ZNF99      0.179160  0.232013    2.0                NaN NaN NaN   \n",
       "ZSCAN12   -0.035386  0.113249    4.0                NaN NaN NaN   \n",
       "ZSCAN12L1 -0.081045       NaN    1.0                NaN NaN NaN   \n",
       "psiTPTE22 -0.085944  0.097844    4.0                NaN NaN NaN   \n",
       "tAKR      -0.059410  0.082184    9.0                NaN NaN NaN   \n",
       "\n",
       "          norm_count_D7_Rep2         norm_count_D14_Rep1 ...  \\\n",
       "                      median std len              median ...   \n",
       "A2BP1                    NaN NaN NaN                 NaN ...   \n",
       "A3GALT2                  NaN NaN NaN                 NaN ...   \n",
       "AACSL                    NaN NaN NaN                 NaN ...   \n",
       "AADACL1                  NaN NaN NaN                 NaN ...   \n",
       "ABCA11P                  NaN NaN NaN                 NaN ...   \n",
       "ABCA17P                  NaN NaN NaN                 NaN ...   \n",
       "ABCC13                   NaN NaN NaN                 NaN ...   \n",
       "ABCC6P1                  NaN NaN NaN                 NaN ...   \n",
       "ABO                      NaN NaN NaN                 NaN ...   \n",
       "ACCN1                    NaN NaN NaN                 NaN ...   \n",
       "ACCN2                    NaN NaN NaN                 NaN ...   \n",
       "ACCN4                    NaN NaN NaN                 NaN ...   \n",
       "ACCN5                    NaN NaN NaN                 NaN ...   \n",
       "ACTL7A                   NaN NaN NaN                 NaN ...   \n",
       "ACTL7B                   NaN NaN NaN                 NaN ...   \n",
       "ACTN3                    NaN NaN NaN                 NaN ...   \n",
       "ACTR2                    NaN NaN NaN                 NaN ...   \n",
       "ADAM3A                   NaN NaN NaN                 NaN ...   \n",
       "ADAM5P                   NaN NaN NaN                 NaN ...   \n",
       "ADFP                     NaN NaN NaN                 NaN ...   \n",
       "ADH1C                    NaN NaN NaN                 NaN ...   \n",
       "ADRA2C                   NaN NaN NaN                 NaN ...   \n",
       "AFG3L1                   NaN NaN NaN                 NaN ...   \n",
       "AFM                      NaN NaN NaN                 NaN ...   \n",
       "AGAP11                   NaN NaN NaN                 NaN ...   \n",
       "AHCTF1                   NaN NaN NaN                 NaN ...   \n",
       "AHDC1                    NaN NaN NaN                 NaN ...   \n",
       "AIM1L                    NaN NaN NaN                 NaN ...   \n",
       "AK3L1                    NaN NaN NaN                 NaN ...   \n",
       "AKAP11                   NaN NaN NaN                 NaN ...   \n",
       "...                      ...  ..  ..                 ... ...   \n",
       "ZNF676                   NaN NaN NaN                 NaN ...   \n",
       "ZNF678                   NaN NaN NaN                 NaN ...   \n",
       "ZNF701                   NaN NaN NaN                 NaN ...   \n",
       "ZNF702P                  NaN NaN NaN                 NaN ...   \n",
       "ZNF706                   NaN NaN NaN                 NaN ...   \n",
       "ZNF716                   NaN NaN NaN                 NaN ...   \n",
       "ZNF717                   NaN NaN NaN                 NaN ...   \n",
       "ZNF718                   NaN NaN NaN                 NaN ...   \n",
       "ZNF721                   NaN NaN NaN                 NaN ...   \n",
       "ZNF735                   NaN NaN NaN                 NaN ...   \n",
       "ZNF738                   NaN NaN NaN                 NaN ...   \n",
       "ZNF761                   NaN NaN NaN                 NaN ...   \n",
       "ZNF763                   NaN NaN NaN                 NaN ...   \n",
       "ZNF765                   NaN NaN NaN                 NaN ...   \n",
       "ZNF767                   NaN NaN NaN                 NaN ...   \n",
       "ZNF788                   NaN NaN NaN                 NaN ...   \n",
       "ZNF80                    NaN NaN NaN                 NaN ...   \n",
       "ZNF815                   NaN NaN NaN                 NaN ...   \n",
       "ZNF816A                  NaN NaN NaN                 NaN ...   \n",
       "ZNF826                   NaN NaN NaN                 NaN ...   \n",
       "ZNF828                   NaN NaN NaN                 NaN ...   \n",
       "ZNF830                   NaN NaN NaN                 NaN ...   \n",
       "ZNF833                   NaN NaN NaN                 NaN ...   \n",
       "ZNF837                   NaN NaN NaN                 NaN ...   \n",
       "ZNF860                   NaN NaN NaN                 NaN ...   \n",
       "ZNF99                    NaN NaN NaN                 NaN ...   \n",
       "ZSCAN12                  NaN NaN NaN                 NaN ...   \n",
       "ZSCAN12L1                NaN NaN NaN                 NaN ...   \n",
       "psiTPTE22                NaN NaN NaN                 NaN ...   \n",
       "tAKR                     NaN NaN NaN                 NaN ...   \n",
       "\n",
       "          norm_count_PLX7_Rep1 norm_count_PLX7_Rep2          \\\n",
       "                           len               median std len   \n",
       "A2BP1                      NaN                  NaN NaN NaN   \n",
       "A3GALT2                    NaN                  NaN NaN NaN   \n",
       "AACSL                      NaN                  NaN NaN NaN   \n",
       "AADACL1                    NaN                  NaN NaN NaN   \n",
       "ABCA11P                    NaN                  NaN NaN NaN   \n",
       "ABCA17P                    NaN                  NaN NaN NaN   \n",
       "ABCC13                     NaN                  NaN NaN NaN   \n",
       "ABCC6P1                    NaN                  NaN NaN NaN   \n",
       "ABO                        NaN                  NaN NaN NaN   \n",
       "ACCN1                      NaN                  NaN NaN NaN   \n",
       "ACCN2                      NaN                  NaN NaN NaN   \n",
       "ACCN4                      NaN                  NaN NaN NaN   \n",
       "ACCN5                      NaN                  NaN NaN NaN   \n",
       "ACTL7A                     NaN                  NaN NaN NaN   \n",
       "ACTL7B                     NaN                  NaN NaN NaN   \n",
       "ACTN3                      NaN                  NaN NaN NaN   \n",
       "ACTR2                      NaN                  NaN NaN NaN   \n",
       "ADAM3A                     NaN                  NaN NaN NaN   \n",
       "ADAM5P                     NaN                  NaN NaN NaN   \n",
       "ADFP                       NaN                  NaN NaN NaN   \n",
       "ADH1C                      NaN                  NaN NaN NaN   \n",
       "ADRA2C                     NaN                  NaN NaN NaN   \n",
       "AFG3L1                     NaN                  NaN NaN NaN   \n",
       "AFM                        NaN                  NaN NaN NaN   \n",
       "AGAP11                     NaN                  NaN NaN NaN   \n",
       "AHCTF1                     NaN                  NaN NaN NaN   \n",
       "AHDC1                      NaN                  NaN NaN NaN   \n",
       "AIM1L                      NaN                  NaN NaN NaN   \n",
       "AK3L1                      NaN                  NaN NaN NaN   \n",
       "AKAP11                     NaN                  NaN NaN NaN   \n",
       "...                        ...                  ...  ..  ..   \n",
       "ZNF676                     NaN                  NaN NaN NaN   \n",
       "ZNF678                     NaN                  NaN NaN NaN   \n",
       "ZNF701                     NaN                  NaN NaN NaN   \n",
       "ZNF702P                    NaN                  NaN NaN NaN   \n",
       "ZNF706                     NaN                  NaN NaN NaN   \n",
       "ZNF716                     NaN                  NaN NaN NaN   \n",
       "ZNF717                     NaN                  NaN NaN NaN   \n",
       "ZNF718                     NaN                  NaN NaN NaN   \n",
       "ZNF721                     NaN                  NaN NaN NaN   \n",
       "ZNF735                     NaN                  NaN NaN NaN   \n",
       "ZNF738                     NaN                  NaN NaN NaN   \n",
       "ZNF761                     NaN                  NaN NaN NaN   \n",
       "ZNF763                     NaN                  NaN NaN NaN   \n",
       "ZNF765                     NaN                  NaN NaN NaN   \n",
       "ZNF767                     NaN                  NaN NaN NaN   \n",
       "ZNF788                     NaN                  NaN NaN NaN   \n",
       "ZNF80                      NaN                  NaN NaN NaN   \n",
       "ZNF815                     NaN                  NaN NaN NaN   \n",
       "ZNF816A                    NaN                  NaN NaN NaN   \n",
       "ZNF826                     NaN                  NaN NaN NaN   \n",
       "ZNF828                     NaN                  NaN NaN NaN   \n",
       "ZNF830                     NaN                  NaN NaN NaN   \n",
       "ZNF833                     NaN                  NaN NaN NaN   \n",
       "ZNF837                     NaN                  NaN NaN NaN   \n",
       "ZNF860                     NaN                  NaN NaN NaN   \n",
       "ZNF99                      NaN                  NaN NaN NaN   \n",
       "ZSCAN12                    NaN                  NaN NaN NaN   \n",
       "ZSCAN12L1                  NaN                  NaN NaN NaN   \n",
       "psiTPTE22                  NaN                  NaN NaN NaN   \n",
       "tAKR                       NaN                  NaN NaN NaN   \n",
       "\n",
       "          norm_count_PLX14_Rep1         norm_count_PLX14_Rep2          \n",
       "                         median std len                median std len  \n",
       "A2BP1                       NaN NaN NaN                   NaN NaN NaN  \n",
       "A3GALT2                     NaN NaN NaN                   NaN NaN NaN  \n",
       "AACSL                       NaN NaN NaN                   NaN NaN NaN  \n",
       "AADACL1                     NaN NaN NaN                   NaN NaN NaN  \n",
       "ABCA11P                     NaN NaN NaN                   NaN NaN NaN  \n",
       "ABCA17P                     NaN NaN NaN                   NaN NaN NaN  \n",
       "ABCC13                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ABCC6P1                     NaN NaN NaN                   NaN NaN NaN  \n",
       "ABO                         NaN NaN NaN                   NaN NaN NaN  \n",
       "ACCN1                       NaN NaN NaN                   NaN NaN NaN  \n",
       "ACCN2                       NaN NaN NaN                   NaN NaN NaN  \n",
       "ACCN4                       NaN NaN NaN                   NaN NaN NaN  \n",
       "ACCN5                       NaN NaN NaN                   NaN NaN NaN  \n",
       "ACTL7A                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ACTL7B                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ACTN3                       NaN NaN NaN                   NaN NaN NaN  \n",
       "ACTR2                       NaN NaN NaN                   NaN NaN NaN  \n",
       "ADAM3A                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ADAM5P                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ADFP                        NaN NaN NaN                   NaN NaN NaN  \n",
       "ADH1C                       NaN NaN NaN                   NaN NaN NaN  \n",
       "ADRA2C                      NaN NaN NaN                   NaN NaN NaN  \n",
       "AFG3L1                      NaN NaN NaN                   NaN NaN NaN  \n",
       "AFM                         NaN NaN NaN                   NaN NaN NaN  \n",
       "AGAP11                      NaN NaN NaN                   NaN NaN NaN  \n",
       "AHCTF1                      NaN NaN NaN                   NaN NaN NaN  \n",
       "AHDC1                       NaN NaN NaN                   NaN NaN NaN  \n",
       "AIM1L                       NaN NaN NaN                   NaN NaN NaN  \n",
       "AK3L1                       NaN NaN NaN                   NaN NaN NaN  \n",
       "AKAP11                      NaN NaN NaN                   NaN NaN NaN  \n",
       "...                         ...  ..  ..                   ...  ..  ..  \n",
       "ZNF676                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF678                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF701                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF702P                     NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF706                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF716                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF717                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF718                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF721                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF735                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF738                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF761                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF763                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF765                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF767                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF788                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF80                       NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF815                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF816A                     NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF826                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF828                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF830                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF833                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF837                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF860                      NaN NaN NaN                   NaN NaN NaN  \n",
       "ZNF99                       NaN NaN NaN                   NaN NaN NaN  \n",
       "ZSCAN12                     NaN NaN NaN                   NaN NaN NaN  \n",
       "ZSCAN12L1                   NaN NaN NaN                   NaN NaN NaN  \n",
       "psiTPTE22                   NaN NaN NaN                   NaN NaN NaN  \n",
       "tAKR                        NaN NaN NaN                   NaN NaN NaN  \n",
       "\n",
       "[2540 rows x 27 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Which genes are not in the guide-seq data?\n",
    "cnv_and_guides[numpy.isnan(cnv_and_guides['norm_count_D7_Rep1']['median'])]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Anyway, we want the inner join - things that are in both sets:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "cnv_and_guides = pandas.merge(cnv_summary_by_gene, guide_abundance, left_index=True, right_index=True, how=\"inner\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sample_column_medians = [(\"CNV\", \"median\"), (\"CNV\", \"std\")]\n",
    "sample_columns.extend([name for index, name in enumerate(column_names) if index in range(0, len(column_names), 3) and name !=\"CNV\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('CNV', 'median'),\n",
       " ('CNV', 'std'),\n",
       " 'norm_count_D7_Rep1',\n",
       " 'norm_count_D7_Rep2',\n",
       " 'norm_count_D14_Rep1',\n",
       " 'norm_count_D14_Rep2',\n",
       " 'norm_count_PLX7_Rep1',\n",
       " 'norm_count_PLX7_Rep2',\n",
       " 'norm_count_PLX14_Rep1',\n",
       " 'norm_count_PLX14_Rep2']"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# We need a broadly applicable way of getting the median guide-seq data for each sample for arbitrary numbers of\n",
    "# arbitrarily named samples\n",
    "column_names = cnv_and_guides.columns.get_level_values(0)\n",
    "sample_column_medians = [(\"CNV\", \"median\"), (\"CNV\", \"std\")]\n",
    "\n",
    "# Take every third item in the list that's not CNV data\n",
    "sample_columns = [name for index, name in enumerate(column_names) if index in range(0, len(column_names), 3) and name !=\"CNV\"]\n",
    "sample_column_medians.extend([(item, \"median\") for item in sample_columns])\n",
    "\n",
    "# Trim the data to the medians of the guide-seq data for each gene\n",
    "df = cnv_and_guides.loc[:,sample_column_medians]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"2\" halign=\"left\">CNV</th>\n",
       "      <th>norm_count_D7_Rep1</th>\n",
       "      <th>norm_count_D7_Rep2</th>\n",
       "      <th>norm_count_D14_Rep1</th>\n",
       "      <th>norm_count_D14_Rep2</th>\n",
       "      <th>norm_count_PLX7_Rep1</th>\n",
       "      <th>norm_count_PLX7_Rep2</th>\n",
       "      <th>norm_count_PLX14_Rep1</th>\n",
       "      <th>norm_count_PLX14_Rep2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>median</th>\n",
       "      <th>std</th>\n",
       "      <th>median</th>\n",
       "      <th>median</th>\n",
       "      <th>median</th>\n",
       "      <th>median</th>\n",
       "      <th>median</th>\n",
       "      <th>median</th>\n",
       "      <th>median</th>\n",
       "      <th>median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1-Dec</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.096822</td>\n",
       "      <td>-0.313602</td>\n",
       "      <td>-0.156710</td>\n",
       "      <td>0.060945</td>\n",
       "      <td>-0.031475</td>\n",
       "      <td>-0.142697</td>\n",
       "      <td>-0.738297</td>\n",
       "      <td>-0.649533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.001451</td>\n",
       "      <td>-0.007358</td>\n",
       "      <td>0.144568</td>\n",
       "      <td>0.079794</td>\n",
       "      <td>-0.095076</td>\n",
       "      <td>-0.160655</td>\n",
       "      <td>-1.105906</td>\n",
       "      <td>-1.053001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.250745</td>\n",
       "      <td>-0.154311</td>\n",
       "      <td>-0.389285</td>\n",
       "      <td>-0.067865</td>\n",
       "      <td>-0.340402</td>\n",
       "      <td>-0.164021</td>\n",
       "      <td>-1.263459</td>\n",
       "      <td>-1.215216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10-Mar</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.316717</td>\n",
       "      <td>0.290522</td>\n",
       "      <td>0.143517</td>\n",
       "      <td>0.317977</td>\n",
       "      <td>0.043548</td>\n",
       "      <td>-0.014089</td>\n",
       "      <td>-1.155326</td>\n",
       "      <td>-0.987062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10-Sep</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.541840</td>\n",
       "      <td>-0.460078</td>\n",
       "      <td>-0.547001</td>\n",
       "      <td>-0.612278</td>\n",
       "      <td>-0.437158</td>\n",
       "      <td>-0.197123</td>\n",
       "      <td>-0.800630</td>\n",
       "      <td>-1.109213</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          CNV     norm_count_D7_Rep1 norm_count_D7_Rep2 norm_count_D14_Rep1  \\\n",
       "       median std             median             median              median   \n",
       "1-Dec     NaN NaN          -0.096822          -0.313602           -0.156710   \n",
       "1-Mar     NaN NaN          -0.001451          -0.007358            0.144568   \n",
       "1-Sep     NaN NaN          -0.250745          -0.154311           -0.389285   \n",
       "10-Mar    NaN NaN           0.316717           0.290522            0.143517   \n",
       "10-Sep    NaN NaN          -0.541840          -0.460078           -0.547001   \n",
       "\n",
       "       norm_count_D14_Rep2 norm_count_PLX7_Rep1 norm_count_PLX7_Rep2  \\\n",
       "                    median               median               median   \n",
       "1-Dec             0.060945            -0.031475            -0.142697   \n",
       "1-Mar             0.079794            -0.095076            -0.160655   \n",
       "1-Sep            -0.067865            -0.340402            -0.164021   \n",
       "10-Mar            0.317977             0.043548            -0.014089   \n",
       "10-Sep           -0.612278            -0.437158            -0.197123   \n",
       "\n",
       "       norm_count_PLX14_Rep1 norm_count_PLX14_Rep2  \n",
       "                      median                median  \n",
       "1-Dec              -0.738297             -0.649533  \n",
       "1-Mar              -1.105906             -1.053001  \n",
       "1-Sep              -1.263459             -1.215216  \n",
       "10-Mar             -1.155326             -0.987062  \n",
       "10-Sep             -0.800630             -1.109213  "
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df.columns = [\"cnv_median\", \"cnv_stdev\", \"d7r1\", \"d7r2\", \"d14r1\", \"d14r2\", \"plx7r1\", \"plx7r2\",\\\n",
    "              \"plx14r1\", \"plx14r2\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "So, graphically compare CNV to enrichment quantile:\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import numpy as np \n",
    "import pylab \n",
    "import scipy.stats as stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df.iloc[:,0].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "quantd = pandas.DataFrame({\"cnv\": df.cnv_median.values, \"plasmidenrichment\": df.d14r2.values})\n",
    "quantd = quantd.sort_values(\"cnv\")\n",
    "nquants = 2000\n",
    "q = quantd.quantile([x/nquants for x in range(0,nquants,1)]).plasmidenrichment\n",
    "labs = pandas.cut(quantd.plasmidenrichment, q)\n",
    "labs.name=\"valbin\"\n",
    "quantd = quantd.join(pandas.DataFrame(labs))\n",
    "binned = quantd.groupby(\"valbin\").median()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "quantd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# What is the actual mean CNV in the total dataset? \n",
    "# Null hypothesis is that each bin of plasmid enrichment values will have average CNV of\n",
    "# the population average\n",
    "df.iloc[:,0].mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%pylab inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pylab.rcParams['figure.figsize'] = (100, 25)\n",
    "quantd.boxplot(\"cnv\", \"valbin\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://connor-johnson.com/2014/02/18/linear-regression-with-python/\n",
    "\n",
    "Try a set of tests on df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import scipy\n",
    "#Pearson correlation coefficient\n",
    "Y = df.d14r2\n",
    "X = df.iloc[:,0]\n",
    "scipy.stats.pearsonr(X,Y)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Correlation is not strong: nearly all CNV is averaged out once you group by guide retention.\n",
    "\n",
    "Genes with no guide selection have highest CNV - doesn't make a difference. So, no selection when there's extra genes around. OK. Positive selection in general is associated with higher \n",
    "\n",
    "So, you get slight enrichment of guides retained as you increase copy number. This makes sense because most knockouts are going to be damaging, and having extra copies will help provide resistance to the sgRNAs.\n",
    "\n",
    "Expect that overall this difference is muted because we're looking at CNV rather than absolute copy number. Guide scores are likely against single copies, though...\n",
    "\n",
    "Now try for other timepoints; expect strengthening of this effect.\n",
    "\n",
    "BRAF should be an outlier?\n",
    "\n",
    "The null hypothesis is no relationship; compare to that for statistical test?\n",
    "\n",
    "CNV trend hypotheses:\n",
    "    Expect BRAF\n",
    "    Oncogenes expanded\n",
    "    Tumor suppressors depleted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# What guide-targeted regions of the genome exhibit the most extreme CNV?\n",
    "df.iloc[:,1].sort_values()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "from scipy.optimize import curve_fit\n",
    "def func(x, a, b, c):\n",
    "    return a * np.exp(-b * x) + c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "xdata = np.linspace(0, 4, 50)\n",
    "y = func(xdata, 2.5, 1.3, 0.5)\n",
    "ydata = y + 0.2 * np.random.normal(size=len(xdata))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "popt, pcov = curve_fit(func, xdata, ydata)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "popt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pcov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "x = df.cnv_median\n",
    "y = df.d7r1\n",
    "e = df.cnv_stdev\n",
    "plt.errorbar(x, y, yerr=e, fmt=\"none\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def line(x, a, b):\n",
    "    return a * x + b\n",
    "\n",
    "popt, pcov = curve_fit(line, x, y, sigma=e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pcov"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "One hypothesis is that extra copies (high CNV) might on average result in less guide selection, either positive or negative. Low CNV might result in increased guide selection.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "%pylab inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "x = df[[\"plx14r1\", \"plx14r2\"]].apply(abs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "y = df.cnv_median\n",
    "e = df.cnv_stdev\n",
    "# plt.scatter(x, y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "x.iloc[:,0].name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "array(df.iloc[:,2:])[:,0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df.iloc[:,2].name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "len(df.iloc[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "figure, axes = plt.subplots(2,4, sharex=True, sharey=True, figsize=(15,15))\n",
    "y = df.cnv_median\n",
    "\n",
    "#for i in axes.ravel():\n",
    "for i, axes in enumerate(axes.ravel()):\n",
    "    x = array(df.iloc[:,2:])[:,i]\n",
    "    print(x)\n",
    "    name = df.iloc[:,2+i].name\n",
    "    axes.hist2d(x, y, bins = 200)\n",
    "    axes.set_title(name)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "array(df.iloc[:,2:])[:,0]"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "array(df.iloc[:,2:])[:,i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "fig, axes =  plt.subplots(nrows=2, ncols=1, sharex=True, sharey=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "popt, pcov = curve_fit(line, x, y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "plt.hist2d(x, y, bins=20, cmap=\"magma\")\n",
    "plt.colorbar()\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "scipy.stats.linregress(x, y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "scipy.stats.linregress(x, y)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
